{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking in indexes: https://repo.huaweicloud.com/repository/pypi/simple\n",
      "Processing /root/autodl-tmp/DeepFRI-CAFA\n",
      "\u001b[33m  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.\n",
      "   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.\u001b[0m\n",
      "\u001b[31mERROR: Could not install packages due to an OSError: [('/root/autodl-tmp/DeepFRI-CAFA/submission.tsv', '/tmp/pip-req-build-0l6z2p8y/submission.tsv', \"[Errno 28] No space left on device: '/root/autodl-tmp/DeepFRI-CAFA/submission.tsv' -> '/tmp/pip-req-build-0l6z2p8y/submission.tsv'\"), ('/root/autodl-tmp/DeepFRI-CAFA/testsuperset.fasta', '/tmp/pip-req-build-0l6z2p8y/testsuperset.fasta', \"[Errno 122] Disk quota exceeded: '/tmp/pip-req-build-0l6z2p8y/testsuperset.fasta'\"), ('/root/autodl-tmp/DeepFRI-CAFA/train_DeepCNN.py', '/tmp/pip-req-build-0l6z2p8y/train_DeepCNN.py', \"[Errno 122] Disk quota exceeded: '/tmp/pip-req-build-0l6z2p8y/train_DeepCNN.py'\"), ('/root/autodl-tmp/DeepFRI-CAFA/train_DeepFRI.py', '/tmp/pip-req-build-0l6z2p8y/train_DeepFRI.py', \"[Errno 122] Disk quota exceeded: '/tmp/pip-req-build-0l6z2p8y/train_DeepFRI.py'\"), ('/root/autodl-tmp/DeepFRI-CAFA/trained_models.tar.gz', '/tmp/pip-req-build-0l6z2p8y/trained_models.tar.gz', \"[Errno 122] Disk quota exceeded: '/tmp/pip-req-build-0l6z2p8y/trained_models.tar.gz'\"), ('/root/autodl-tmp/DeepFRI-CAFA/viz_gradCAM.py', '/tmp/pip-req-build-0l6z2p8y/viz_gradCAM.py', \"[Errno 122] Disk quota exceeded: '/tmp/pip-req-build-0l6z2p8y/viz_gradCAM.py'\")]\n",
      "\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip install ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      " DATA DIRECTORY (./data/) CREATED!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.chdir('./preprocessing')\n",
    "\n",
    "DATA_DIR = './data/'\n",
    "TFR_DIR = './data/ContactMaps/TFRecords'\n",
    "SEQ_SIM = 95\n",
    "\n",
    "os.makedirs(DATA_DIR, exist_ok=True)\n",
    "print(f\"\\n\\n DATA DIRECTORY ({DATA_DIR}) CREATED!\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"\\n\\n DOWNLOADING SIFTS-GO DATA...\\n\")\n",
    "!wget ftp://ftp.ebi.ac.uk/pub/databases/msd/sifts/flatfiles/tsv/pdb_chain_go.tsv.gz -O ./data/pdb_chain_go.tsv.gz\n",
    "\n",
    "print(\"\\n\\n DOWNLOADING SIFTS-EC DATA...\\n\")\n",
    "!wget ftp://ftp.ebi.ac.uk/pub/databases/msd/sifts/flatfiles/tsv/pdb_chain_enzyme.tsv.gz -O ./data/pdb_chain_enzyme.tsv.gz\n",
    "\n",
    "print(\"\\n\\n DOWNLOADING PDB SEQRES SEQUENCES...\\n\")\n",
    "!wget ftp://ftp.wwpdb.org/pub/pdb/derived_data/pdb_seqres.txt.gz -O ./data/pdb_seqres.txt.gz\n",
    "\n",
    "# print(\"\\n\\n DOWNLOADING PDB CLUSTERS...\\n\")\n",
    "# !wget https://cdn.rcsb.org/resources/sequence/clusters/clusters-by-entity-95.txt -O ./data/bc-95.out\n",
    "\n",
    "# print(\"\\n\\n DOWNLOADING GO HIERARCHY...\\n\")\n",
    "!wget http://purl.obolibrary.org/obo/go/go-basic.obo -O ./data/go-basic.obo\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "python PDB2distMap.py -annot ./data/CAFA__annot.tsv -seqres ./data/pdb_seqres.txt.gz -num_threads 30 -bc ./data/bc-95.out -out_dir ./data/annot_pdb_chains_npz/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "  PREPROCESSING GO-ANNOTATIONS [Please wait this process may take a few minutes]...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(\"\\n\\n  PREPROCESSING GO-ANNOTATIONS [Please wait this process may take a few minutes]...\\n\")\n",
    "!python create_nrPDB_GO_annot.py -sifts ./data/pdb_chain_go.tsv.gz -bc ./data/bc-95.out -seqres ./data/pdb_seqres.txt.gz -obo ./data/go-basic.obo -out ./data/CAFA_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ./data/ContactMaps/\n",
    "!mkdir ./data/ContactMaps/TFRecords/\n",
    "!python PDB2TFRecord.py -annot ./data/CAFA__annot.tsv -prot_list ./data/CAFA__train.txt -npz_dir ./data/annot_pdb_chains_npz/ -num_shards 30 -num_threads 30 -tfr_prefix ./data/ContactMaps/TFRecords/PDB_GO_train\n",
    "!python PDB2TFRecord.py -annot ./data/CAFA__annot.tsv -prot_list ./data/CAFA__valid.txt -npz_dir ./data/annot_pdb_chains_npz/ -num_shards 3 -num_threads 3 -tfr_prefix ./data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-06-01 21:35:34.317691: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-01 21:35:35.290039: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-01 21:35:35.326969: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-01 21:35:35.327495: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:06.0 name: Tesla V100S-PCIE-32GB computeCapability: 7.0\n",
      "coreClock: 1.597GHz coreCount: 80 deviceMemorySize: 31.74GiB deviceMemoryBandwidth: 1.03TiB/s\n",
      "2023-06-01 21:35:35.327523: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-01 21:35:35.346156: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-01 21:35:35.346219: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-01 21:35:35.356034: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-01 21:35:35.359766: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-01 21:35:35.364462: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-01 21:35:35.368850: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-01 21:35:35.369077: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-01 21:35:35.369202: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-01 21:35:35.369721: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-01 21:35:35.370841: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n"
     ]
    }
   ],
   "source": [
    "import tensorflow as tf\n",
    "gpus = tf.config.experimental.list_physical_devices('GPU')\n",
    "if gpus:\n",
    "    try:\n",
    "        for gpu in gpus:\n",
    "            tf.config.experimental.set_memory_growth(gpu, True)\n",
    "    except RuntimeError as e:\n",
    "        print(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-18 21:43:29.721539: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 21:43:31.376753: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-18 21:43:31.488385: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:31.490313: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 21:43:31.490363: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 21:43:31.494197: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 21:43:31.494306: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 21:43:31.495520: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-18 21:43:31.495911: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-18 21:43:31.496979: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-18 21:43:31.497849: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-18 21:43:31.498075: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 21:43:31.498232: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:31.500076: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:31.501834: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "Namespace(annot_fn='./preprocessing/data/CAFA2_annot.tsv', batch_size=64, dropout=0.3, epochs=60, filter_lens=[8, 16, 25, 32, 40, 48, 56, 64, 72, 80, 88, 96, 104, 112, 120, 128], l2_reg=0.0002, lm_model_name=None, lr=0.0002, model_name='./trained_models/DeepCNN_BP', num_filters=[256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256], ontology='bp', pad_len=None, test_list='./preprocessing/data/nrPDB-GO_2019.06.18_test.csv', train_tfrecord_fn='./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train', valid_tfrecord_fn='./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid')\n",
      "### Training model:  ./trained_models/DeepCNN_BP  on  978  GO terms.\n",
      "### Compiling DeepCNN model...\n",
      "2023-06-18 21:43:32.713163: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-06-18 21:43:32.718954: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:32.720994: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 21:43:32.721204: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:32.723064: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:32.724810: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-18 21:43:32.724907: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 21:43:33.619401: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-06-18 21:43:33.619447: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 \n",
      "2023-06-18 21:43:33.619460: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N \n",
      "2023-06-18 21:43:33.620084: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:33.621916: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:33.623812: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 21:43:33.625325: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38286 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:00:07.0, compute capability: 8.0)\n",
      "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py:374: UserWarning: The `lr` argument is deprecated, use `learning_rate` instead.\n",
      "  warnings.warn(\n",
      "Model: \"model\"\n",
      "__________________________________________________________________________________________________\n",
      "Layer (type)                    Output Shape         Param #     Connected to                     \n",
      "==================================================================================================\n",
      "seq (InputLayer)                [(None, None, 26)]   0                                            \n",
      "__________________________________________________________________________________________________\n",
      "conv1d (Conv1D)                 (None, None, 256)    53504       seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_1 (Conv1D)               (None, None, 256)    106752      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_2 (Conv1D)               (None, None, 256)    166656      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_3 (Conv1D)               (None, None, 256)    213248      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_4 (Conv1D)               (None, None, 256)    266496      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_5 (Conv1D)               (None, None, 256)    319744      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_6 (Conv1D)               (None, None, 256)    372992      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_7 (Conv1D)               (None, None, 256)    426240      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_8 (Conv1D)               (None, None, 256)    479488      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_9 (Conv1D)               (None, None, 256)    532736      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_10 (Conv1D)              (None, None, 256)    585984      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_11 (Conv1D)              (None, None, 256)    639232      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_12 (Conv1D)              (None, None, 256)    692480      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_13 (Conv1D)              (None, None, 256)    745728      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_14 (Conv1D)              (None, None, 256)    798976      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_15 (Conv1D)              (None, None, 256)    852224      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "concatenate (Concatenate)       (None, None, 4096)   0           conv1d[0][0]                     \n",
      "                                                                 conv1d_1[0][0]                   \n",
      "                                                                 conv1d_2[0][0]                   \n",
      "                                                                 conv1d_3[0][0]                   \n",
      "                                                                 conv1d_4[0][0]                   \n",
      "                                                                 conv1d_5[0][0]                   \n",
      "                                                                 conv1d_6[0][0]                   \n",
      "                                                                 conv1d_7[0][0]                   \n",
      "                                                                 conv1d_8[0][0]                   \n",
      "                                                                 conv1d_9[0][0]                   \n",
      "                                                                 conv1d_10[0][0]                  \n",
      "                                                                 conv1d_11[0][0]                  \n",
      "                                                                 conv1d_12[0][0]                  \n",
      "                                                                 conv1d_13[0][0]                  \n",
      "                                                                 conv1d_14[0][0]                  \n",
      "                                                                 conv1d_15[0][0]                  \n",
      "__________________________________________________________________________________________________\n",
      "batch_normalization (BatchNorma (None, None, 4096)   16384       concatenate[0][0]                \n",
      "__________________________________________________________________________________________________\n",
      "CNN_concatenate (Activation)    (None, None, 4096)   0           batch_normalization[0][0]        \n",
      "__________________________________________________________________________________________________\n",
      "dropout (Dropout)               (None, None, 4096)   0           CNN_concatenate[0][0]            \n",
      "__________________________________________________________________________________________________\n",
      "global_max_pooling1d (GlobalMax (None, 4096)         0           dropout[0][0]                    \n",
      "__________________________________________________________________________________________________\n",
      "dropout_1 (Dropout)             (None, 4096)         0           global_max_pooling1d[0][0]       \n",
      "__________________________________________________________________________________________________\n",
      "labels (FuncPredictor)          (None, 978, 2)       8013732     dropout_1[0][0]                  \n",
      "==================================================================================================\n",
      "Total params: 15,282,596\n",
      "Trainable params: 15,274,404\n",
      "Non-trainable params: 8,192\n",
      "__________________________________________________________________________________________________\n",
      "None\n",
      "2023-06-18 21:43:34.007669: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:176] None of the MLIR Optimization Passes are enabled (registered 2)\n",
      "2023-06-18 21:43:34.009126: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2999890000 Hz\n",
      "### Training on:  1926 contact maps.\n",
      "### Validating on:  214 contact maps.\n",
      "Epoch 1/60\n",
      "2023-06-18 21:43:38.737822: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 21:43:48.086342: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8101\n",
      "2023-06-18 21:43:52.231069: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 21:43:54.213818: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 21:43:55.238826: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "30/30 [==============================] - 90s 2s/step - loss: 1.3131 - acc: 0.9225 - val_loss: 0.3548 - val_acc: 0.9608\n",
      "\n",
      "Epoch 00001: val_loss improved from inf to 0.35482, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 2/60\n",
      "30/30 [==============================] - 28s 943ms/step - loss: 0.7687 - acc: 0.9470 - val_loss: 0.3886 - val_acc: 0.9640\n",
      "\n",
      "Epoch 00002: val_loss did not improve from 0.35482\n",
      "Epoch 3/60\n",
      "30/30 [==============================] - 16s 541ms/step - loss: 0.5303 - acc: 0.9489 - val_loss: 0.4073 - val_acc: 0.9656\n",
      "\n",
      "Epoch 00003: val_loss did not improve from 0.35482\n",
      "Epoch 4/60\n",
      "30/30 [==============================] - 14s 426ms/step - loss: 0.4206 - acc: 0.9510 - val_loss: 0.4065 - val_acc: 0.9689\n",
      "\n",
      "Epoch 00004: val_loss did not improve from 0.35482\n",
      "Epoch 5/60\n",
      "30/30 [==============================] - 5s 161ms/step - loss: 0.3379 - acc: 0.9570 - val_loss: 0.3757 - val_acc: 0.9699\n",
      "\n",
      "Epoch 00005: val_loss did not improve from 0.35482\n",
      "Epoch 6/60\n",
      "30/30 [==============================] - 11s 319ms/step - loss: 0.2962 - acc: 0.9610 - val_loss: 0.3462 - val_acc: 0.9717\n",
      "\n",
      "Epoch 00006: val_loss improved from 0.35482 to 0.34623, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 7/60\n",
      "30/30 [==============================] - 11s 356ms/step - loss: 0.2542 - acc: 0.9658 - val_loss: 0.3119 - val_acc: 0.9729\n",
      "\n",
      "Epoch 00007: val_loss improved from 0.34623 to 0.31186, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 8/60\n",
      "30/30 [==============================] - 5s 161ms/step - loss: 0.2319 - acc: 0.9691 - val_loss: 0.2802 - val_acc: 0.9741\n",
      "\n",
      "Epoch 00008: val_loss improved from 0.31186 to 0.28018, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 9/60\n",
      "30/30 [==============================] - 8s 257ms/step - loss: 0.2129 - acc: 0.9712 - val_loss: 0.2536 - val_acc: 0.9752\n",
      "\n",
      "Epoch 00009: val_loss improved from 0.28018 to 0.25356, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 10/60\n",
      "30/30 [==============================] - 10s 297ms/step - loss: 0.1902 - acc: 0.9752 - val_loss: 0.2239 - val_acc: 0.9776\n",
      "\n",
      "Epoch 00010: val_loss improved from 0.25356 to 0.22390, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 11/60\n",
      "30/30 [==============================] - 6s 200ms/step - loss: 0.1755 - acc: 0.9763 - val_loss: 0.1993 - val_acc: 0.9780\n",
      "\n",
      "Epoch 00011: val_loss improved from 0.22390 to 0.19931, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 12/60\n",
      "30/30 [==============================] - 5s 161ms/step - loss: 0.1647 - acc: 0.9777 - val_loss: 0.1828 - val_acc: 0.9781\n",
      "\n",
      "Epoch 00012: val_loss improved from 0.19931 to 0.18276, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 13/60\n",
      "30/30 [==============================] - 10s 333ms/step - loss: 0.1496 - acc: 0.9803 - val_loss: 0.1649 - val_acc: 0.9798\n",
      "\n",
      "Epoch 00013: val_loss improved from 0.18276 to 0.16492, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 14/60\n",
      "30/30 [==============================] - 4s 147ms/step - loss: 0.1398 - acc: 0.9813 - val_loss: 0.1510 - val_acc: 0.9804\n",
      "\n",
      "Epoch 00014: val_loss improved from 0.16492 to 0.15103, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 15/60\n",
      "30/30 [==============================] - 5s 154ms/step - loss: 0.1228 - acc: 0.9843 - val_loss: 0.1368 - val_acc: 0.9814\n",
      "\n",
      "Epoch 00015: val_loss improved from 0.15103 to 0.13680, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 16/60\n",
      "30/30 [==============================] - 8s 210ms/step - loss: 0.1179 - acc: 0.9843 - val_loss: 0.1313 - val_acc: 0.9818\n",
      "\n",
      "Epoch 00016: val_loss improved from 0.13680 to 0.13135, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 17/60\n",
      "30/30 [==============================] - 5s 159ms/step - loss: 0.1070 - acc: 0.9862 - val_loss: 0.1249 - val_acc: 0.9821\n",
      "\n",
      "Epoch 00017: val_loss improved from 0.13135 to 0.12492, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 18/60\n",
      "30/30 [==============================] - 6s 190ms/step - loss: 0.1018 - acc: 0.9865 - val_loss: 0.1208 - val_acc: 0.9826\n",
      "\n",
      "Epoch 00018: val_loss improved from 0.12492 to 0.12075, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 19/60\n",
      "30/30 [==============================] - 9s 293ms/step - loss: 0.0947 - acc: 0.9875 - val_loss: 0.1210 - val_acc: 0.9821\n",
      "\n",
      "Epoch 00019: val_loss did not improve from 0.12075\n",
      "Epoch 20/60\n",
      "30/30 [==============================] - 4s 141ms/step - loss: 0.0862 - acc: 0.9890 - val_loss: 0.1139 - val_acc: 0.9840\n",
      "\n",
      "Epoch 00020: val_loss improved from 0.12075 to 0.11389, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 21/60\n",
      "30/30 [==============================] - 6s 162ms/step - loss: 0.0827 - acc: 0.9889 - val_loss: 0.1025 - val_acc: 0.9854\n",
      "\n",
      "Epoch 00021: val_loss improved from 0.11389 to 0.10252, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 22/60\n",
      "30/30 [==============================] - 7s 227ms/step - loss: 0.0766 - acc: 0.9900 - val_loss: 0.1022 - val_acc: 0.9855\n",
      "\n",
      "Epoch 00022: val_loss improved from 0.10252 to 0.10219, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 23/60\n",
      "30/30 [==============================] - 5s 164ms/step - loss: 0.0728 - acc: 0.9905 - val_loss: 0.1103 - val_acc: 0.9836\n",
      "\n",
      "Epoch 00023: val_loss did not improve from 0.10219\n",
      "Epoch 24/60\n",
      "30/30 [==============================] - 5s 155ms/step - loss: 0.0680 - acc: 0.9913 - val_loss: 0.1064 - val_acc: 0.9844\n",
      "\n",
      "Epoch 00024: val_loss did not improve from 0.10219\n",
      "Epoch 25/60\n",
      "30/30 [==============================] - 8s 264ms/step - loss: 0.0659 - acc: 0.9912 - val_loss: 0.1100 - val_acc: 0.9828\n",
      "\n",
      "Epoch 00025: val_loss did not improve from 0.10219\n",
      "Epoch 26/60\n",
      "30/30 [==============================] - 4s 150ms/step - loss: 0.0623 - acc: 0.9918 - val_loss: 0.1061 - val_acc: 0.9834\n",
      "\n",
      "Epoch 00026: val_loss did not improve from 0.10219\n",
      "Epoch 27/60\n",
      "30/30 [==============================] - 5s 157ms/step - loss: 0.0591 - acc: 0.9922 - val_loss: 0.0960 - val_acc: 0.9856\n",
      "\n",
      "Epoch 00027: val_loss improved from 0.10219 to 0.09603, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 28/60\n",
      "30/30 [==============================] - 5s 170ms/step - loss: 0.0568 - acc: 0.9927 - val_loss: 0.0952 - val_acc: 0.9860\n",
      "\n",
      "Epoch 00028: val_loss improved from 0.09603 to 0.09524, saving model to ./trained_models/DeepCNN_BP_best_train_model.h5\n",
      "Epoch 29/60\n",
      "30/30 [==============================] - 6s 196ms/step - loss: 0.0553 - acc: 0.9925 - val_loss: 0.1049 - val_acc: 0.9839\n",
      "\n",
      "Epoch 00029: val_loss did not improve from 0.09524\n",
      "Epoch 30/60\n",
      "30/30 [==============================] - 5s 160ms/step - loss: 0.0535 - acc: 0.9929 - val_loss: 0.1026 - val_acc: 0.9841\n",
      "\n",
      "Epoch 00030: val_loss did not improve from 0.09524\n",
      "Epoch 31/60\n",
      "30/30 [==============================] - 6s 197ms/step - loss: 0.0518 - acc: 0.9933 - val_loss: 0.1074 - val_acc: 0.9838\n",
      "\n",
      "Epoch 00031: val_loss did not improve from 0.09524\n",
      "Epoch 32/60\n",
      "30/30 [==============================] - 4s 133ms/step - loss: 0.0499 - acc: 0.9936 - val_loss: 0.1033 - val_acc: 0.9835\n",
      "\n",
      "Epoch 00032: val_loss did not improve from 0.09524\n",
      "Epoch 33/60\n",
      "30/30 [==============================] - 4s 142ms/step - loss: 0.0474 - acc: 0.9941 - val_loss: 0.0970 - val_acc: 0.9852\n",
      "\n",
      "Epoch 00033: val_loss did not improve from 0.09524\n"
     ]
    }
   ],
   "source": [
    "!python train_DeepCNN.py -ont bp -bs 64 --model_name ./trained_models/DeepCNN_BP --annot_fn ./preprocessing/data/CAFA2_annot.tsv --train_tfrecord_fn ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train --valid_tfrecord_fn ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Namespace(annot_fn='./preprocessing/data/CAFA2_annot.tsv', batch_size=1, cmap_thresh=10.0, cmap_type='ca', dropout=0.3, epochs=200, fc_dims=[2048], gc_dims=[512, 512, 512], gc_layer='MultiGraphConv', l2_reg=2e-05, lm_model_name='./trained_models/lstm_lm_tf.hdf5', lr=0.0002, model_name='./trained_models/test', ontology='bp', pad_len=None, test_list='./preprocessing/data/nrPDB-GO_2019.06.18_test.csv', train_tfrecord_fn='./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train', valid_tfrecord_fn='./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid')\n",
      "### Training model:  ./trained_models/test  on  3941  GO terms.\n",
      "### Compiling DeepFRI model with MultiGraphConv layer...\n",
      "Model: \"functional_3\"\n",
      "__________________________________________________________________________________________________\n",
      "Layer (type)                    Output Shape         Param #     Connected to                     \n",
      "==================================================================================================\n",
      "seq (InputLayer)                [(None, None, 26)]   0                                            \n",
      "__________________________________________________________________________________________________\n",
      "functional_1 (Functional)       (None, None, 1024)   3203072     seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "LM_embedding (Dense)            (None, None, 1024)   1049600     functional_1[0][0]               \n",
      "__________________________________________________________________________________________________\n",
      "AA_embedding (Dense)            (None, None, 1024)   26624       seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "Embedding (Add)                 (None, None, 1024)   0           LM_embedding[0][0]               \n",
      "                                                                 AA_embedding[0][0]               \n",
      "__________________________________________________________________________________________________\n",
      "activation (Activation)         (None, None, 1024)   0           Embedding[0][0]                  \n",
      "__________________________________________________________________________________________________\n",
      "cmap (InputLayer)               [(None, None, None)] 0                                            \n",
      "__________________________________________________________________________________________________\n",
      "MultiGraphConv_1 (MultiGraphCon (None, None, 512)    1572864     activation[0][0]                 \n",
      "                                                                 cmap[0][0]                       \n",
      "__________________________________________________________________________________________________\n",
      "MultiGraphConv_2 (MultiGraphCon (None, None, 512)    786432      MultiGraphConv_1[0][0]           \n",
      "                                                                 cmap[0][0]                       \n",
      "__________________________________________________________________________________________________\n",
      "MultiGraphConv_3 (MultiGraphCon (None, None, 512)    786432      MultiGraphConv_2[0][0]           \n",
      "                                                                 cmap[0][0]                       \n",
      "__________________________________________________________________________________________________\n",
      "GCNN_concatenate (Concatenate)  (None, None, 1536)   0           MultiGraphConv_1[0][0]           \n",
      "                                                                 MultiGraphConv_2[0][0]           \n",
      "                                                                 MultiGraphConv_3[0][0]           \n",
      "__________________________________________________________________________________________________\n",
      "SumPooling (SumPooling)         (None, 1536)         0           GCNN_concatenate[0][0]           \n",
      "__________________________________________________________________________________________________\n",
      "dense (Dense)                   (None, 2048)         3147776     SumPooling[0][0]                 \n",
      "__________________________________________________________________________________________________\n",
      "dropout (Dropout)               (None, 2048)         0           dense[0][0]                      \n",
      "__________________________________________________________________________________________________\n",
      "labels (FuncPredictor)          (None, 3941, 2)      16150218    dropout[0][0]                    \n",
      "==================================================================================================\n",
      "Total params: 26,723,018\n",
      "Trainable params: 23,519,946\n",
      "Non-trainable params: 3,203,072\n",
      "__________________________________________________________________________________________________\n",
      "None\n",
      "???\n",
      "### Training on:  41 contact maps.\n",
      "### Validating on:  5 contact maps.\n",
      "Epoch 1/200\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-07-02 09:43:59.475020: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cudart64_101.dll'; dlerror: cudart64_101.dll not found\n",
      "2023-07-02 09:43:59.475244: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.\n",
      "2023-07-02 09:44:02.659049: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library nvcuda.dll\n",
      "2023-07-02 09:44:02.679522: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: \n",
      "pciBusID: 0000:01:00.0 name: NVIDIA GeForce RTX 3060 Laptop GPU computeCapability: 8.6\n",
      "coreClock: 1.425GHz coreCount: 30 deviceMemorySize: 6.00GiB deviceMemoryBandwidth: 312.97GiB/s\n",
      "2023-07-02 09:44:02.684632: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cudart64_101.dll'; dlerror: cudart64_101.dll not found\n",
      "2023-07-02 09:44:02.689848: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cublas64_10.dll'; dlerror: cublas64_10.dll not found\n",
      "2023-07-02 09:44:02.695805: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cufft64_10.dll\n",
      "2023-07-02 09:44:02.697278: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library curand64_10.dll\n",
      "2023-07-02 09:44:02.702500: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cusolver64_10.dll'; dlerror: cusolver64_10.dll not found\n",
      "2023-07-02 09:44:02.707304: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cusparse64_10.dll'; dlerror: cusparse64_10.dll not found\n",
      "2023-07-02 09:44:02.712283: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cudnn64_7.dll'; dlerror: cudnn64_7.dll not found\n",
      "2023-07-02 09:44:02.712480: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1753] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.\n",
      "Skipping registering GPU devices...\n",
      "2023-07-02 09:44:05.040769: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-07-02 09:44:05.047885: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x1c2f8b8f050 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n",
      "2023-07-02 09:44:05.048277: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n",
      "2023-07-02 09:44:05.048486: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1257] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-07-02 09:44:05.048601: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1263]      \n",
      "2023-07-02 09:44:09.038130: W tensorflow/core/framework/op_kernel.cc:1767] OP_REQUIRES failed at example_parsing_ops.cc:94 : Invalid argument: Key: bp_labels.  Can't parse serialized Example.\n",
      "Traceback (most recent call last):\n",
      "  File \"train_DeepFRI.py\", line 71, in <module>\n",
      "    cmap_type=args.cmap_type, cmap_thresh=args.cmap_thresh, ont=args.ontology, class_weight=None)\n",
      "  File \"i:\\DeepFRI-CAFA\\deepfrier\\DeepFRI.py\", line 153, in train\n",
      "    callbacks=[es, mc])\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\keras\\engine\\training.py\", line 108, in _method_wrapper\n",
      "    return method(self, *args, **kwargs)\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\keras\\engine\\training.py\", line 1098, in fit\n",
      "    tmp_logs = train_function(iterator)\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\def_function.py\", line 780, in __call__\n",
      "    result = self._call(*args, **kwds)\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\def_function.py\", line 840, in _call\n",
      "    return self._stateless_fn(*args, **kwds)\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\function.py\", line 2829, in __call__\n",
      "    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\function.py\", line 1848, in _filtered_call\n",
      "    cancellation_manager=cancellation_manager)\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\function.py\", line 1924, in _call_flat\n",
      "    ctx, args, cancellation_manager=cancellation_manager))\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\function.py\", line 550, in call\n",
      "    ctx=ctx)\n",
      "  File \"e:\\Envs\\deepfri\\lib\\site-packages\\tensorflow\\python\\eager\\execute.py\", line 60, in quick_execute\n",
      "    inputs, attrs, num_outputs)\n",
      "tensorflow.python.framework.errors_impl.InvalidArgumentError:  Key: bp_labels.  Can't parse serialized Example.\n",
      "\t [[{{node ParseSingleExample/ParseExample/ParseExampleV2}}]]\n",
      "\t [[IteratorGetNext]] [Op:__inference_train_function_8999]\n",
      "\n",
      "Function call stack:\n",
      "train_function\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!python train_DeepFRI.py -gc MultiGraphConv -bs 1 -ont bp --model_name ./trained_models/test --annot_fn ./preprocessing/data/CAFA2_annot.tsv -tf ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train -vf ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-18 23:21:57.851143: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:21:59.597603: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-18 23:21:59.744172: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:21:59.746085: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 23:21:59.746147: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:21:59.752261: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 23:21:59.752367: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 23:21:59.754355: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-18 23:21:59.756474: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-18 23:21:59.759472: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-18 23:21:59.760387: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-18 23:21:59.760619: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 23:21:59.760776: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:21:59.762617: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:21:59.764319: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "Namespace(annot_fn='./preprocessing/data/CAFA2_annot.tsv', batch_size=64, dropout=0.3, epochs=60, filter_lens=[8, 16, 25, 32, 40, 48, 56, 64, 72, 80, 88, 96, 104, 112, 120, 128], l2_reg=0.0002, lm_model_name=None, lr=0.0002, model_name='./trained_models/DeepCNN_CC', num_filters=[256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256, 256], ontology='cc', pad_len=None, test_list='./preprocessing/data/nrPDB-GO_2019.06.18_test.csv', train_tfrecord_fn='./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train', valid_tfrecord_fn='./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid')\n",
      "### Training model:  ./trained_models/DeepCNN_CC  on  645  GO terms.\n",
      "### Compiling DeepCNN model...\n",
      "2023-06-18 23:22:00.975842: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-06-18 23:22:00.982343: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:22:00.984352: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 23:22:00.984563: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:22:00.986328: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:22:00.988041: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-18 23:22:00.988114: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:22:01.879889: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-06-18 23:22:01.879937: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 \n",
      "2023-06-18 23:22:01.879952: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N \n",
      "2023-06-18 23:22:01.880235: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:22:01.882218: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:22:01.884001: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:22:01.885525: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38286 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:00:07.0, compute capability: 8.0)\n",
      "/root/miniconda3/lib/python3.8/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py:374: UserWarning: The `lr` argument is deprecated, use `learning_rate` instead.\n",
      "  warnings.warn(\n",
      "Model: \"model\"\n",
      "__________________________________________________________________________________________________\n",
      "Layer (type)                    Output Shape         Param #     Connected to                     \n",
      "==================================================================================================\n",
      "seq (InputLayer)                [(None, None, 26)]   0                                            \n",
      "__________________________________________________________________________________________________\n",
      "conv1d (Conv1D)                 (None, None, 256)    53504       seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_1 (Conv1D)               (None, None, 256)    106752      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_2 (Conv1D)               (None, None, 256)    166656      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_3 (Conv1D)               (None, None, 256)    213248      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_4 (Conv1D)               (None, None, 256)    266496      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_5 (Conv1D)               (None, None, 256)    319744      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_6 (Conv1D)               (None, None, 256)    372992      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_7 (Conv1D)               (None, None, 256)    426240      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_8 (Conv1D)               (None, None, 256)    479488      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_9 (Conv1D)               (None, None, 256)    532736      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_10 (Conv1D)              (None, None, 256)    585984      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_11 (Conv1D)              (None, None, 256)    639232      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_12 (Conv1D)              (None, None, 256)    692480      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_13 (Conv1D)              (None, None, 256)    745728      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_14 (Conv1D)              (None, None, 256)    798976      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "conv1d_15 (Conv1D)              (None, None, 256)    852224      seq[0][0]                        \n",
      "__________________________________________________________________________________________________\n",
      "concatenate (Concatenate)       (None, None, 4096)   0           conv1d[0][0]                     \n",
      "                                                                 conv1d_1[0][0]                   \n",
      "                                                                 conv1d_2[0][0]                   \n",
      "                                                                 conv1d_3[0][0]                   \n",
      "                                                                 conv1d_4[0][0]                   \n",
      "                                                                 conv1d_5[0][0]                   \n",
      "                                                                 conv1d_6[0][0]                   \n",
      "                                                                 conv1d_7[0][0]                   \n",
      "                                                                 conv1d_8[0][0]                   \n",
      "                                                                 conv1d_9[0][0]                   \n",
      "                                                                 conv1d_10[0][0]                  \n",
      "                                                                 conv1d_11[0][0]                  \n",
      "                                                                 conv1d_12[0][0]                  \n",
      "                                                                 conv1d_13[0][0]                  \n",
      "                                                                 conv1d_14[0][0]                  \n",
      "                                                                 conv1d_15[0][0]                  \n",
      "__________________________________________________________________________________________________\n",
      "batch_normalization (BatchNorma (None, None, 4096)   16384       concatenate[0][0]                \n",
      "__________________________________________________________________________________________________\n",
      "CNN_concatenate (Activation)    (None, None, 4096)   0           batch_normalization[0][0]        \n",
      "__________________________________________________________________________________________________\n",
      "dropout (Dropout)               (None, None, 4096)   0           CNN_concatenate[0][0]            \n",
      "__________________________________________________________________________________________________\n",
      "global_max_pooling1d (GlobalMax (None, 4096)         0           dropout[0][0]                    \n",
      "__________________________________________________________________________________________________\n",
      "dropout_1 (Dropout)             (None, 4096)         0           global_max_pooling1d[0][0]       \n",
      "__________________________________________________________________________________________________\n",
      "labels (FuncPredictor)          (None, 645, 2)       5285130     dropout_1[0][0]                  \n",
      "==================================================================================================\n",
      "Total params: 12,553,994\n",
      "Trainable params: 12,545,802\n",
      "Non-trainable params: 8,192\n",
      "__________________________________________________________________________________________________\n",
      "None\n",
      "2023-06-18 23:22:02.380425: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:176] None of the MLIR Optimization Passes are enabled (registered 2)\n",
      "2023-06-18 23:22:02.397407: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2999890000 Hz\n",
      "### Training on:  1926 contact maps.\n",
      "### Validating on:  214 contact maps.\n",
      "Epoch 1/60\n",
      "2023-06-18 23:22:07.193417: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 23:22:08.088781: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8101\n",
      "2023-06-18 23:22:09.035900: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 23:22:09.744743: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 23:22:10.378370: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "30/30 [==============================] - 67s 2s/step - loss: 1.1960 - acc: 0.9322 - val_loss: 0.2880 - val_acc: 0.9721\n",
      "\n",
      "Epoch 00001: val_loss improved from inf to 0.28803, saving model to ./trained_models/DeepCNN_CC_best_train_model.hdf5\n",
      "Epoch 2/60\n",
      "30/30 [==============================] - 29s 978ms/step - loss: 0.6966 - acc: 0.9603 - val_loss: 0.2962 - val_acc: 0.9746\n",
      "\n",
      "Epoch 00002: val_loss did not improve from 0.28803\n",
      "Epoch 3/60\n",
      " 8/30 [=======>......................] - ETA: 12s - loss: 0.5573 - acc: 0.9600"
     ]
    }
   ],
   "source": [
    "!python train_DeepCNN.py -ont cc -bs 64 --model_name ./trained_models/DeepCNN_CC --annot_fn ./preprocessing/data/CAFA2_annot.tsv --train_tfrecord_fn ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train --valid_tfrecord_fn ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python train_DeepFRI.py -gc MultiGraphConv -ont cc --model_name ./trained_models/MultiGCN_CC --annot_fn ./preprocessing/data/CAFA2_annot.tsv -tf ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train -vf ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python train_DeepCNN.py -ont mf -bs 64 --model_name ./trained_models/DeepCNN_MF --annot_fn ./preprocessing/data/CAFA2_annot.tsv --train_tfrecord_fn ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train --valid_tfrecord_fn ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python train_DeepFRI.py -gc MultiGraphConv -ont mf --model_name ./trained_models/MultiGCN_MF --annot_fn ./preprocessing/data/CAFA2_annot.tsv -tf ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_train -vf ./preprocessing/data/ContactMaps/TFRecords/PDB_GO_valid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-18 23:18:41.168104: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:18:42.602110: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-18 23:18:42.747756: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:42.749665: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 23:18:42.749711: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:18:42.755850: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 23:18:42.755939: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 23:18:42.760773: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-18 23:18:42.761132: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-18 23:18:42.764844: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-18 23:18:42.766856: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-18 23:18:42.767073: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 23:18:42.767213: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:42.768991: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:42.770697: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-18 23:18:42.771158: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-06-18 23:18:42.776363: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:42.778081: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 23:18:42.778197: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:42.779982: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:42.781669: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-18 23:18:42.781730: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:18:43.668536: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-06-18 23:18:43.668589: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 \n",
      "2023-06-18 23:18:43.668603: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N \n",
      "2023-06-18 23:18:43.669364: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:43.671223: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:43.672758: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:18:43.674245: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38286 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:00:07.0, compute capability: 8.0)\n",
      "### Computing predictions on a single protein...\n",
      "2023-06-18 23:18:45.126883: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 23:18:45.977378: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8101\n",
      "2023-06-18 23:18:47.110199: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 23:18:47.866689: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 23:18:48.080710: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "Protein GO-term/EC-number Score GO-term/EC-number name\n",
      "query_prot GO:0009987 0.91350 cellular process\n",
      "query_prot GO:0044237 0.83579 cellular metabolic process\n",
      "query_prot GO:0008150 0.83514 biological_process\n",
      "query_prot GO:0008152 0.62488 metabolic process\n",
      "query_prot GO:0044238 0.54546 primary metabolic process\n",
      "query_prot GO:0065007 0.44922 biological regulation\n",
      "query_prot GO:0006807 0.43269 nitrogen compound metabolic process\n",
      "query_prot GO:0071704 0.42735 organic substance metabolic process\n",
      "query_prot GO:0010467 0.41923 gene expression\n",
      "query_prot GO:0034641 0.40299 cellular nitrogen compound metabolic process\n",
      "query_prot GO:0050789 0.37735 regulation of biological process\n",
      "query_prot GO:0044249 0.34850 cellular biosynthetic process\n",
      "query_prot GO:0071840 0.33853 cellular component organization or biogenesis\n",
      "query_prot GO:1901360 0.33514 organic cyclic compound metabolic process\n",
      "query_prot GO:0046483 0.30157 heterocycle metabolic process\n",
      "query_prot GO:0043170 0.30068 macromolecule metabolic process\n",
      "query_prot GO:0050794 0.29069 regulation of cellular process\n",
      "query_prot GO:0044271 0.27395 cellular nitrogen compound biosynthetic process\n",
      "query_prot GO:0006725 0.26697 cellular aromatic compound metabolic process\n",
      "query_prot GO:0009059 0.26012 macromolecule biosynthetic process\n",
      "query_prot GO:0009058 0.24815 biosynthetic process\n",
      "query_prot GO:0006139 0.24420 nucleobase-containing compound metabolic process\n",
      "query_prot GO:0090304 0.23148 nucleic acid metabolic process\n",
      "query_prot GO:0050896 0.17520 response to stimulus\n",
      "query_prot GO:0032501 0.14916 multicellular organismal process\n",
      "query_prot GO:1901362 0.14619 organic cyclic compound biosynthetic process\n",
      "query_prot GO:0051179 0.13728 localization\n",
      "query_prot GO:0060255 0.12425 regulation of macromolecule metabolic process\n",
      "query_prot GO:0009893 0.12378 positive regulation of metabolic process\n",
      "query_prot GO:0032502 0.12285 developmental process\n",
      "query_prot GO:0019438 0.11750 aromatic compound biosynthetic process\n",
      "query_prot GO:0016070 0.11721 RNA metabolic process\n",
      "query_prot GO:0048518 0.11521 positive regulation of biological process\n",
      "query_prot GO:1901564 0.11141 organonitrogen compound metabolic process\n",
      "query_prot GO:1901576 0.11007 organic substance biosynthetic process\n",
      "query_prot GO:0031323 0.10912 regulation of cellular metabolic process\n",
      "query_prot GO:0018130 0.10861 heterocycle biosynthetic process\n",
      "query_prot GO:0006950 0.10839 response to stress\n",
      "query_prot GO:0022414 0.10572 reproductive process\n",
      "query_prot GO:0006810 0.10411 transport\n",
      "query_prot GO:0048856 0.10386 anatomical structure development\n",
      "query_prot GO:0051171 0.10240 regulation of nitrogen compound metabolic process\n",
      "query_prot GO:0044085 0.09730 cellular component biogenesis\n",
      "query_prot GO:0048522 0.09200 positive regulation of cellular process\n",
      "query_prot GO:0007275 0.09046 multicellular organism development\n",
      "query_prot GO:0016043 0.09031 cellular component organization\n",
      "query_prot GO:0034654 0.08405 nucleobase-containing compound biosynthetic process\n",
      "query_prot GO:0032774 0.08110 RNA biosynthetic process\n",
      "query_prot GO:0019538 0.07636 protein metabolic process\n",
      "query_prot GO:0010468 0.06963 regulation of gene expression\n",
      "query_prot GO:0033554 0.06952 cellular response to stress\n",
      "query_prot GO:0048869 0.06557 cellular developmental process\n",
      "query_prot GO:0051716 0.06302 cellular response to stimulus\n",
      "query_prot GO:0080090 0.06092 regulation of primary metabolic process\n",
      "query_prot GO:0044281 0.05798 small molecule metabolic process\n",
      "query_prot GO:0000003 0.05629 reproduction\n",
      "query_prot GO:0019222 0.04853 regulation of metabolic process\n",
      "query_prot GO:0043933 0.04584 protein-containing complex organization\n",
      "query_prot GO:0006913 0.04443 nucleocytoplasmic transport\n",
      "query_prot GO:0006518 0.04305 peptide metabolic process\n",
      "query_prot GO:0006996 0.04285 organelle organization\n",
      "query_prot GO:0051234 0.03918 establishment of localization\n",
      "query_prot GO:0010604 0.03885 positive regulation of macromolecule metabolic process\n",
      "query_prot GO:0043604 0.03865 amide biosynthetic process\n",
      "query_prot GO:0012501 0.03654 programmed cell death\n",
      "query_prot GO:0031326 0.03584 regulation of cellular biosynthetic process\n",
      "query_prot GO:0030154 0.03581 cell differentiation\n",
      "query_prot GO:0071705 0.03504 nitrogen compound transport\n",
      "query_prot GO:0048731 0.03488 system development\n",
      "query_prot GO:0022607 0.03484 cellular component assembly\n",
      "query_prot GO:0042221 0.03463 response to chemical\n",
      "query_prot GO:0009653 0.03419 anatomical structure morphogenesis\n",
      "query_prot GO:0008104 0.03190 protein localization\n",
      "query_prot GO:0007399 0.03173 nervous system development\n",
      "query_prot GO:0044419 0.03123 biological process involved in interspecies interaction between organisms\n",
      "query_prot GO:0009605 0.03060 response to external stimulus\n",
      "query_prot GO:0043603 0.03054 amide metabolic process\n",
      "query_prot GO:0022402 0.03012 cell cycle process\n",
      "query_prot GO:0043412 0.02998 macromolecule modification\n",
      "query_prot GO:0006796 0.02952 phosphate-containing compound metabolic process\n",
      "query_prot GO:0019219 0.02949 regulation of nucleobase-containing compound metabolic process\n",
      "query_prot GO:0048513 0.02911 animal organ development\n",
      "query_prot GO:0051701 0.02773 biological process involved in interaction with host\n",
      "query_prot GO:0045184 0.02760 establishment of protein localization\n",
      "query_prot GO:1901566 0.02728 organonitrogen compound biosynthetic process\n",
      "query_prot GO:0051252 0.02566 regulation of RNA metabolic process\n",
      "query_prot GO:0010033 0.02558 response to organic substance\n",
      "query_prot GO:0031325 0.02547 positive regulation of cellular metabolic process\n",
      "query_prot GO:0006979 0.02532 response to oxidative stress\n",
      "query_prot GO:0048519 0.02529 negative regulation of biological process\n",
      "query_prot GO:0009889 0.02511 regulation of biosynthetic process\n",
      "query_prot GO:0008219 0.02426 cell death\n",
      "query_prot GO:0070727 0.02403 cellular macromolecule localization\n",
      "query_prot GO:0065003 0.02400 protein-containing complex assembly\n",
      "query_prot GO:0006974 0.02370 DNA damage response\n",
      "query_prot GO:0009607 0.02350 response to biotic stimulus\n",
      "query_prot GO:0010556 0.02344 regulation of macromolecule biosynthetic process\n",
      "query_prot GO:0023052 0.02336 signaling\n",
      "query_prot GO:0051649 0.02298 establishment of localization in cell\n",
      "query_prot GO:0006915 0.02245 apoptotic process\n",
      "query_prot GO:0036211 0.02227 protein modification process\n",
      "query_prot GO:0043067 0.02127 regulation of programmed cell death\n",
      "query_prot GO:0006412 0.02122 translation\n",
      "query_prot GO:0071702 0.02100 organic substance transport\n",
      "query_prot GO:1901700 0.02069 response to oxygen-containing compound\n",
      "query_prot GO:0033036 0.02001 macromolecule localization\n",
      "query_prot GO:0015980 0.01982 energy derivation by oxidation of organic compounds\n",
      "query_prot GO:0009892 0.01916 negative regulation of metabolic process\n",
      "query_prot GO:0006793 0.01907 phosphorus metabolic process\n",
      "query_prot GO:0006366 0.01902 transcription by RNA polymerase II\n",
      "query_prot GO:0003006 0.01863 developmental process involved in reproduction\n",
      "query_prot GO:0031667 0.01853 response to nutrient levels\n",
      "query_prot GO:0048468 0.01842 cell development\n",
      "query_prot GO:0007049 0.01812 cell cycle\n",
      "query_prot GO:0051173 0.01781 positive regulation of nitrogen compound metabolic process\n",
      "query_prot GO:0051052 0.01746 regulation of DNA metabolic process\n",
      "query_prot GO:0045333 0.01740 cellular respiration\n",
      "query_prot GO:0009628 0.01711 response to abiotic stimulus\n",
      "query_prot GO:0007276 0.01680 gamete generation\n",
      "query_prot GO:0048870 0.01622 cell motility\n",
      "query_prot GO:0070482 0.01613 response to oxygen levels\n",
      "query_prot GO:0016071 0.01600 mRNA metabolic process\n",
      "query_prot GO:1901575 0.01596 organic substance catabolic process\n",
      "query_prot GO:0044087 0.01559 regulation of cellular component biogenesis\n",
      "query_prot GO:0002376 0.01557 immune system process\n",
      "query_prot GO:2001141 0.01505 regulation of RNA biosynthetic process\n",
      "query_prot GO:0015031 0.01494 protein transport\n",
      "query_prot GO:0019953 0.01472 sexual reproduction\n",
      "query_prot GO:0032504 0.01471 multicellular organism reproduction\n",
      "query_prot GO:0051321 0.01393 meiotic cell cycle\n",
      "query_prot GO:0006403 0.01382 RNA localization\n",
      "query_prot GO:0065009 0.01362 regulation of molecular function\n",
      "query_prot GO:0051239 0.01358 regulation of multicellular organismal process\n",
      "query_prot GO:0051641 0.01357 cellular localization\n",
      "query_prot GO:0051240 0.01351 positive regulation of multicellular organismal process\n",
      "query_prot GO:0070887 0.01338 cellular response to chemical stimulus\n",
      "query_prot GO:0019058 0.01330 viral life cycle\n",
      "query_prot GO:0000278 0.01324 mitotic cell cycle\n",
      "query_prot GO:0048583 0.01311 regulation of response to stimulus\n",
      "query_prot GO:0010605 0.01306 negative regulation of macromolecule metabolic process\n",
      "query_prot GO:0010557 0.01284 positive regulation of macromolecule biosynthetic process\n",
      "query_prot GO:0007281 0.01253 germ cell development\n",
      "query_prot GO:0006259 0.01248 DNA metabolic process\n",
      "query_prot GO:0031324 0.01246 negative regulation of cellular metabolic process\n",
      "query_prot GO:0048523 0.01238 negative regulation of cellular process\n",
      "query_prot GO:0006811 0.01237 monoatomic ion transport\n",
      "query_prot GO:0010628 0.01205 positive regulation of gene expression\n",
      "query_prot GO:0051169 0.01179 nuclear transport\n",
      "query_prot GO:0046907 0.01179 intracellular transport\n",
      "query_prot GO:0003008 0.01177 system process\n",
      "query_prot GO:0006260 0.01160 DNA replication\n",
      "query_prot GO:0072359 0.01147 circulatory system development\n",
      "query_prot GO:0051246 0.01129 regulation of protein metabolic process\n",
      "query_prot GO:0006302 0.01119 double-strand break repair\n",
      "query_prot GO:0051707 0.01108 response to other organism\n",
      "query_prot GO:0006091 0.01104 generation of precursor metabolites and energy\n",
      "query_prot GO:0022613 0.01101 ribonucleoprotein complex biogenesis\n",
      "query_prot GO:0006351 0.01097 DNA-templated transcription\n",
      "query_prot GO:0010243 0.01085 response to organonitrogen compound\n",
      "query_prot GO:0048646 0.01054 anatomical structure formation involved in morphogenesis\n",
      "query_prot GO:0009887 0.01043 animal organ morphogenesis\n",
      "query_prot GO:0001932 0.01033 regulation of protein phosphorylation\n",
      "query_prot GO:0009967 0.01029 positive regulation of signal transduction\n",
      "query_prot GO:0016192 0.01028 vesicle-mediated transport\n",
      "query_prot GO:0090066 0.01028 regulation of anatomical structure size\n",
      "query_prot GO:0051726 0.01016 regulation of cell cycle\n"
     ]
    }
   ],
   "source": [
    "# p20536 test\n",
    "!python predict.py --seq MNSVTVSHAPYTITYHDDWEPVMSQLVEFYNEVASWLLRDETSPIPDKFFIQLKQPLRNKRVCVCGIDPYPKDGTGVPFESPNFTKKSIKEIASSISRLTGVIDYKGYNLNIIDGVIPWNYYLSCKLGETKSHAIYWDKISKLLLQHITKHVSVLYCLGKTDFSNIRAKLESPVTTIVGYHPAARDRQFEKDRSFEIINVLLELDNKVPINWAQGFIY -ont bp --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-18 23:13:23.111848: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:13:24.592289: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-18 23:13:24.738367: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:24.740932: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 23:13:24.740997: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:13:24.757997: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 23:13:24.758149: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 23:13:24.771318: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-18 23:13:24.772277: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-18 23:13:24.776450: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-18 23:13:24.781379: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-18 23:13:24.782244: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 23:13:24.783119: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:24.784982: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:24.787957: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-18 23:13:24.793728: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-06-18 23:13:24.800748: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:24.802568: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:07.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-18 23:13:24.802697: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:24.804474: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:24.806167: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-18 23:13:24.806377: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-18 23:13:25.719652: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-06-18 23:13:25.719698: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 \n",
      "2023-06-18 23:13:25.719719: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N \n",
      "2023-06-18 23:13:25.721734: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:25.724169: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:25.726302: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-18 23:13:25.728962: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38286 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:00:07.0, compute capability: 8.0)\n",
      "### Computing predictions on a single protein...\n",
      "2023-06-18 23:13:27.414098: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-18 23:13:28.215827: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8101\n",
      "2023-06-18 23:13:29.243209: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-18 23:13:30.006819: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-18 23:13:30.145860: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "Protein GO-term/EC-number Score GO-term/EC-number name\n",
      "query_prot GO:0009987 0.98888 cellular process\n",
      "query_prot GO:0008150 0.96864 biological_process\n",
      "query_prot GO:0008152 0.90638 metabolic process\n",
      "query_prot GO:0071704 0.87977 organic substance metabolic process\n",
      "query_prot GO:0010467 0.80108 gene expression\n",
      "query_prot GO:0044237 0.79787 cellular metabolic process\n",
      "query_prot GO:0043170 0.79591 macromolecule metabolic process\n",
      "query_prot GO:0044238 0.78796 primary metabolic process\n",
      "query_prot GO:0006807 0.77633 nitrogen compound metabolic process\n",
      "query_prot GO:0071840 0.74979 cellular component organization or biogenesis\n",
      "query_prot GO:0050789 0.66369 regulation of biological process\n",
      "query_prot GO:0016043 0.65820 cellular component organization\n",
      "query_prot GO:0034641 0.60970 cellular nitrogen compound metabolic process\n",
      "query_prot GO:0065007 0.60442 biological regulation\n",
      "query_prot GO:0032502 0.56576 developmental process\n",
      "query_prot GO:0044249 0.53417 cellular biosynthetic process\n",
      "query_prot GO:0050794 0.52919 regulation of cellular process\n",
      "query_prot GO:0006725 0.52126 cellular aromatic compound metabolic process\n",
      "query_prot GO:0009058 0.49361 biosynthetic process\n",
      "query_prot GO:0032501 0.45233 multicellular organismal process\n",
      "query_prot GO:0090304 0.42635 nucleic acid metabolic process\n",
      "query_prot GO:0019538 0.40758 protein metabolic process\n",
      "query_prot GO:0044271 0.39401 cellular nitrogen compound biosynthetic process\n",
      "query_prot GO:0048856 0.39134 anatomical structure development\n",
      "query_prot GO:1901360 0.39105 organic cyclic compound metabolic process\n",
      "query_prot GO:0044085 0.37010 cellular component biogenesis\n",
      "query_prot GO:0048869 0.36514 cellular developmental process\n",
      "query_prot GO:0006139 0.35631 nucleobase-containing compound metabolic process\n",
      "query_prot GO:0009059 0.33325 macromolecule biosynthetic process\n",
      "query_prot GO:0016070 0.33215 RNA metabolic process\n",
      "query_prot GO:0007275 0.32881 multicellular organism development\n",
      "query_prot GO:0022607 0.31193 cellular component assembly\n",
      "query_prot GO:1901564 0.30428 organonitrogen compound metabolic process\n",
      "query_prot GO:0046483 0.27533 heterocycle metabolic process\n",
      "query_prot GO:0051179 0.26610 localization\n",
      "query_prot GO:0060255 0.26582 regulation of macromolecule metabolic process\n",
      "query_prot GO:0043933 0.25743 protein-containing complex organization\n",
      "query_prot GO:1901576 0.23196 organic substance biosynthetic process\n",
      "query_prot GO:0019222 0.23055 regulation of metabolic process\n",
      "query_prot GO:0000003 0.21653 reproduction\n",
      "query_prot GO:0006996 0.20909 organelle organization\n",
      "query_prot GO:0080090 0.20640 regulation of primary metabolic process\n",
      "query_prot GO:0022414 0.19986 reproductive process\n",
      "query_prot GO:0030154 0.18547 cell differentiation\n",
      "query_prot GO:1901362 0.18448 organic cyclic compound biosynthetic process\n",
      "query_prot GO:0051171 0.18303 regulation of nitrogen compound metabolic process\n",
      "query_prot GO:0070727 0.17441 cellular macromolecule localization\n",
      "query_prot GO:0051234 0.16720 establishment of localization\n",
      "query_prot GO:0006950 0.16302 response to stress\n",
      "query_prot GO:0050896 0.16161 response to stimulus\n",
      "query_prot GO:0032774 0.16136 RNA biosynthetic process\n",
      "query_prot GO:0003006 0.16037 developmental process involved in reproduction\n",
      "query_prot GO:0006810 0.16001 transport\n",
      "query_prot GO:0048518 0.14882 positive regulation of biological process\n",
      "query_prot GO:0022402 0.14687 cell cycle process\n",
      "query_prot GO:0048519 0.14499 negative regulation of biological process\n",
      "query_prot GO:0065003 0.14402 protein-containing complex assembly\n",
      "query_prot GO:0034654 0.14098 nucleobase-containing compound biosynthetic process\n",
      "query_prot GO:0048731 0.14081 system development\n",
      "query_prot GO:0010468 0.12910 regulation of gene expression\n",
      "query_prot GO:0051716 0.12534 cellular response to stimulus\n",
      "query_prot GO:0033036 0.12523 macromolecule localization\n",
      "query_prot GO:0031326 0.11948 regulation of cellular biosynthetic process\n",
      "query_prot GO:0007049 0.11072 cell cycle\n",
      "query_prot GO:0045184 0.10866 establishment of protein localization\n",
      "query_prot GO:0006518 0.10623 peptide metabolic process\n",
      "query_prot GO:0018130 0.10611 heterocycle biosynthetic process\n",
      "query_prot GO:2001141 0.10430 regulation of RNA biosynthetic process\n",
      "query_prot GO:0006357 0.10123 regulation of transcription by RNA polymerase II\n",
      "query_prot GO:0048468 0.10092 cell development\n",
      "query_prot GO:0019438 0.10087 aromatic compound biosynthetic process\n",
      "query_prot GO:0008104 0.09081 protein localization\n",
      "query_prot GO:0009893 0.09007 positive regulation of metabolic process\n",
      "query_prot GO:0031323 0.08987 regulation of cellular metabolic process\n",
      "query_prot GO:0009892 0.08485 negative regulation of metabolic process\n",
      "query_prot GO:0048522 0.08450 positive regulation of cellular process\n",
      "query_prot GO:0051641 0.08356 cellular localization\n",
      "query_prot GO:0048523 0.07901 negative regulation of cellular process\n",
      "query_prot GO:0007389 0.07531 pattern specification process\n",
      "query_prot GO:0071705 0.07407 nitrogen compound transport\n",
      "query_prot GO:0033554 0.07210 cellular response to stress\n",
      "query_prot GO:0048285 0.07195 organelle fission\n",
      "query_prot GO:0051649 0.07011 establishment of localization in cell\n",
      "query_prot GO:0010556 0.06927 regulation of macromolecule biosynthetic process\n",
      "query_prot GO:0006351 0.06759 DNA-templated transcription\n",
      "query_prot GO:0007399 0.06612 nervous system development\n",
      "query_prot GO:0007276 0.06588 gamete generation\n",
      "query_prot GO:0071702 0.06586 organic substance transport\n",
      "query_prot GO:0007010 0.06513 cytoskeleton organization\n",
      "query_prot GO:0006366 0.06157 transcription by RNA polymerase II\n",
      "query_prot GO:0019219 0.06105 regulation of nucleobase-containing compound metabolic process\n",
      "query_prot GO:0042254 0.05956 ribosome biogenesis\n",
      "query_prot GO:0000278 0.05918 mitotic cell cycle\n",
      "query_prot GO:0051239 0.05761 regulation of multicellular organismal process\n",
      "query_prot GO:0009653 0.05742 anatomical structure morphogenesis\n",
      "query_prot GO:0048609 0.05611 multicellular organismal reproductive process\n",
      "query_prot GO:0043043 0.05571 peptide biosynthetic process\n",
      "query_prot GO:0048870 0.05434 cell motility\n",
      "query_prot GO:0010604 0.05288 positive regulation of macromolecule metabolic process\n",
      "query_prot GO:0051252 0.05199 regulation of RNA metabolic process\n",
      "query_prot GO:0007017 0.05190 microtubule-based process\n",
      "query_prot GO:0006974 0.05178 DNA damage response\n",
      "query_prot GO:0009889 0.05124 regulation of biosynthetic process\n",
      "query_prot GO:0051246 0.05124 regulation of protein metabolic process\n",
      "query_prot GO:0010557 0.05030 positive regulation of macromolecule biosynthetic process\n",
      "query_prot GO:0009628 0.05013 response to abiotic stimulus\n",
      "query_prot GO:0043604 0.04962 amide biosynthetic process\n",
      "query_prot GO:0051172 0.04909 negative regulation of nitrogen compound metabolic process\n",
      "query_prot GO:0023052 0.04843 signaling\n",
      "query_prot GO:0048583 0.04788 regulation of response to stimulus\n",
      "query_prot GO:1901566 0.04754 organonitrogen compound biosynthetic process\n",
      "query_prot GO:0006355 0.04726 regulation of DNA-templated transcription\n",
      "query_prot GO:0006913 0.04674 nucleocytoplasmic transport\n",
      "query_prot GO:0003008 0.04635 system process\n",
      "query_prot GO:0009790 0.04593 embryo development\n",
      "query_prot GO:0019953 0.04576 sexual reproduction\n",
      "query_prot GO:0032504 0.04550 multicellular organism reproduction\n",
      "query_prot GO:0051248 0.04438 negative regulation of protein metabolic process\n",
      "query_prot GO:0010605 0.04325 negative regulation of macromolecule metabolic process\n",
      "query_prot GO:0048646 0.04281 anatomical structure formation involved in morphogenesis\n",
      "query_prot GO:0006811 0.04256 monoatomic ion transport\n",
      "query_prot GO:0033365 0.04134 protein localization to organelle\n",
      "query_prot GO:1902680 0.04055 positive regulation of RNA biosynthetic process\n",
      "query_prot GO:0042221 0.04000 response to chemical\n",
      "query_prot GO:0009888 0.03955 tissue development\n",
      "query_prot GO:1903047 0.03855 mitotic cell cycle process\n",
      "query_prot GO:0006281 0.03681 DNA repair\n",
      "query_prot GO:0022613 0.03668 ribonucleoprotein complex biogenesis\n",
      "query_prot GO:0043603 0.03619 amide metabolic process\n",
      "query_prot GO:0000226 0.03582 microtubule cytoskeleton organization\n",
      "query_prot GO:0031324 0.03509 negative regulation of cellular metabolic process\n",
      "query_prot GO:0003002 0.03493 regionalization\n",
      "query_prot GO:0006259 0.03429 DNA metabolic process\n",
      "query_prot GO:0008283 0.03385 cell population proliferation\n",
      "query_prot GO:0009605 0.03380 response to external stimulus\n",
      "query_prot GO:0006412 0.03322 translation\n",
      "query_prot GO:0048232 0.03262 male gamete generation\n",
      "query_prot GO:0045893 0.03255 positive regulation of DNA-templated transcription\n",
      "query_prot GO:0009791 0.03237 post-embryonic development\n",
      "query_prot GO:0051173 0.03085 positive regulation of nitrogen compound metabolic process\n",
      "query_prot GO:0006915 0.03022 apoptotic process\n",
      "query_prot GO:0045944 0.03001 positive regulation of transcription by RNA polymerase II\n",
      "query_prot GO:0007281 0.02975 germ cell development\n",
      "query_prot GO:0007283 0.02957 spermatogenesis\n",
      "query_prot GO:0051301 0.02940 cell division\n",
      "query_prot GO:0006396 0.02919 RNA processing\n",
      "query_prot GO:0031325 0.02905 positive regulation of cellular metabolic process\n",
      "query_prot GO:0065009 0.02883 regulation of molecular function\n",
      "query_prot GO:0010033 0.02862 response to organic substance\n",
      "query_prot GO:0048513 0.02859 animal organ development\n",
      "query_prot GO:0051169 0.02849 nuclear transport\n",
      "query_prot GO:0010558 0.02840 negative regulation of macromolecule biosynthetic process\n",
      "query_prot GO:0007154 0.02805 cell communication\n",
      "query_prot GO:0006508 0.02785 proteolysis\n",
      "query_prot GO:1901575 0.02647 organic substance catabolic process\n",
      "query_prot GO:0009887 0.02546 animal organ morphogenesis\n",
      "query_prot GO:0031328 0.02546 positive regulation of cellular biosynthetic process\n",
      "query_prot GO:0016071 0.02479 mRNA metabolic process\n",
      "query_prot GO:0036211 0.02477 protein modification process\n",
      "query_prot GO:0046907 0.02409 intracellular transport\n",
      "query_prot GO:0008219 0.02383 cell death\n",
      "query_prot GO:0070887 0.02379 cellular response to chemical stimulus\n",
      "query_prot GO:0070925 0.02370 organelle assembly\n",
      "query_prot GO:0060322 0.02365 head development\n",
      "query_prot GO:0009056 0.02358 catabolic process\n",
      "query_prot GO:0034660 0.02358 ncRNA metabolic process\n",
      "query_prot GO:0044281 0.02349 small molecule metabolic process\n",
      "query_prot GO:0030097 0.02343 hemopoiesis\n",
      "query_prot GO:0012501 0.02324 programmed cell death\n",
      "query_prot GO:0007507 0.02323 heart development\n",
      "query_prot GO:0051321 0.02272 meiotic cell cycle\n",
      "query_prot GO:0050877 0.02226 nervous system process\n",
      "query_prot GO:0071310 0.02208 cellular response to organic substance\n",
      "query_prot GO:0015031 0.02143 protein transport\n",
      "query_prot GO:0009855 0.02142 determination of bilateral symmetry\n",
      "query_prot GO:0098542 0.02112 defense response to other organism\n",
      "query_prot GO:0019941 0.02103 modification-dependent protein catabolic process\n",
      "query_prot GO:0044248 0.02067 cellular catabolic process\n",
      "query_prot GO:1903046 0.02042 meiotic cell cycle process\n",
      "query_prot GO:0072657 0.02027 protein localization to membrane\n",
      "query_prot GO:0000819 0.02022 sister chromatid segregation\n",
      "query_prot GO:0000280 0.02018 nuclear division\n",
      "query_prot GO:0007059 0.02017 chromosome segregation\n",
      "query_prot GO:0001822 0.01997 kidney development\n",
      "query_prot GO:0006397 0.01995 mRNA processing\n",
      "query_prot GO:0043412 0.01992 macromolecule modification\n",
      "query_prot GO:0032446 0.01969 protein modification by small protein conjugation\n",
      "query_prot GO:0009607 0.01949 response to biotic stimulus\n",
      "query_prot GO:0090068 0.01935 positive regulation of cell cycle process\n",
      "query_prot GO:0042127 0.01919 regulation of cell population proliferation\n",
      "query_prot GO:0010628 0.01916 positive regulation of gene expression\n",
      "query_prot GO:0010638 0.01905 positive regulation of organelle organization\n",
      "query_prot GO:0010564 0.01885 regulation of cell cycle process\n",
      "query_prot GO:0000398 0.01877 mRNA splicing, via spliceosome\n",
      "query_prot GO:0010498 0.01850 proteasomal protein catabolic process\n",
      "query_prot GO:0009891 0.01836 positive regulation of biosynthetic process\n",
      "query_prot GO:0009057 0.01826 macromolecule catabolic process\n",
      "query_prot GO:0006793 0.01822 phosphorus metabolic process\n",
      "query_prot GO:0030900 0.01810 forebrain development\n",
      "query_prot GO:0031667 0.01804 response to nutrient levels\n",
      "query_prot GO:0010243 0.01791 response to organonitrogen compound\n",
      "query_prot GO:0044092 0.01747 negative regulation of molecular function\n",
      "query_prot GO:0030031 0.01731 cell projection assembly\n",
      "query_prot GO:0065008 0.01731 regulation of biological quality\n",
      "query_prot GO:0048608 0.01730 reproductive structure development\n",
      "query_prot GO:0002376 0.01729 immune system process\n",
      "query_prot GO:0048880 0.01691 sensory system development\n",
      "query_prot GO:0045862 0.01690 positive regulation of proteolysis\n",
      "query_prot GO:0062012 0.01686 regulation of small molecule metabolic process\n",
      "query_prot GO:0023056 0.01685 positive regulation of signaling\n",
      "query_prot GO:0050793 0.01676 regulation of developmental process\n",
      "query_prot GO:0010154 0.01668 fruit development\n",
      "query_prot GO:0006310 0.01658 DNA recombination\n",
      "query_prot GO:0042592 0.01656 homeostatic process\n",
      "query_prot GO:0051707 0.01636 response to other organism\n",
      "query_prot GO:0046530 0.01624 photoreceptor cell differentiation\n",
      "query_prot GO:0060429 0.01617 epithelium development\n",
      "query_prot GO:0016072 0.01599 rRNA metabolic process\n",
      "query_prot GO:0000375 0.01547 RNA splicing, via transesterification reactions\n",
      "query_prot GO:0001932 0.01534 regulation of protein phosphorylation\n",
      "query_prot GO:0043086 0.01495 negative regulation of catalytic activity\n",
      "query_prot GO:0009314 0.01487 response to radiation\n",
      "query_prot GO:0035295 0.01483 tube development\n",
      "query_prot GO:0009967 0.01448 positive regulation of signal transduction\n",
      "query_prot GO:0044093 0.01443 positive regulation of molecular function\n",
      "query_prot GO:0044782 0.01433 cilium organization\n",
      "query_prot GO:0015980 0.01421 energy derivation by oxidation of organic compounds\n",
      "query_prot GO:0016192 0.01419 vesicle-mediated transport\n",
      "query_prot GO:0009966 0.01414 regulation of signal transduction\n",
      "query_prot GO:1901700 0.01414 response to oxygen-containing compound\n",
      "query_prot GO:0009719 0.01402 response to endogenous stimulus\n",
      "query_prot GO:0005975 0.01400 carbohydrate metabolic process\n",
      "query_prot GO:1901565 0.01395 organonitrogen compound catabolic process\n",
      "query_prot GO:0007165 0.01386 signal transduction\n",
      "query_prot GO:0071824 0.01383 protein-DNA complex organization\n",
      "query_prot GO:0035556 0.01376 intracellular signal transduction\n",
      "query_prot GO:0009725 0.01364 response to hormone\n",
      "query_prot GO:0007166 0.01363 cell surface receptor signaling pathway\n",
      "query_prot GO:0009799 0.01361 specification of symmetry\n",
      "query_prot GO:0051247 0.01360 positive regulation of protein metabolic process\n",
      "query_prot GO:0048598 0.01351 embryonic morphogenesis\n",
      "query_prot GO:0008285 0.01348 negative regulation of cell population proliferation\n",
      "query_prot GO:0030163 0.01331 protein catabolic process\n",
      "query_prot GO:0140013 0.01330 meiotic nuclear division\n",
      "query_prot GO:0051129 0.01326 negative regulation of cellular component organization\n",
      "query_prot GO:0006952 0.01320 defense response\n",
      "query_prot GO:0042325 0.01320 regulation of phosphorylation\n",
      "query_prot GO:0043085 0.01319 positive regulation of catalytic activity\n",
      "query_prot GO:0032886 0.01318 regulation of microtubule-based process\n",
      "query_prot GO:0150063 0.01318 visual system development\n",
      "query_prot GO:0097435 0.01308 supramolecular fiber organization\n",
      "query_prot GO:0098813 0.01307 nuclear chromosome segregation\n",
      "query_prot GO:0009890 0.01286 negative regulation of biosynthetic process\n",
      "query_prot GO:0090066 0.01283 regulation of anatomical structure size\n",
      "query_prot GO:0001654 0.01283 eye development\n",
      "query_prot GO:0051656 0.01267 establishment of organelle localization\n",
      "query_prot GO:0051726 0.01263 regulation of cell cycle\n",
      "query_prot GO:0006796 0.01256 phosphate-containing compound metabolic process\n",
      "query_prot GO:0010256 0.01239 endomembrane system organization\n",
      "query_prot GO:0060341 0.01239 regulation of cellular localization\n",
      "query_prot GO:0140014 0.01233 mitotic nuclear division\n",
      "query_prot GO:0048316 0.01229 seed development\n",
      "query_prot GO:0072001 0.01227 renal system development\n",
      "query_prot GO:0007420 0.01226 brain development\n",
      "query_prot GO:0051128 0.01222 regulation of cellular component organization\n",
      "query_prot GO:0009792 0.01186 embryo development ending in birth or egg hatching\n",
      "query_prot GO:0007155 0.01182 cell adhesion\n",
      "query_prot GO:0140694 0.01170 non-membrane-bounded organelle assembly\n",
      "query_prot GO:2000241 0.01161 regulation of reproductive process\n",
      "query_prot GO:0006790 0.01150 sulfur compound metabolic process\n",
      "query_prot GO:0098657 0.01145 import into cell\n",
      "query_prot GO:0009615 0.01143 response to virus\n",
      "query_prot GO:0051098 0.01139 regulation of binding\n",
      "query_prot GO:0030334 0.01117 regulation of cell migration\n",
      "query_prot GO:0048585 0.01116 negative regulation of response to stimulus\n",
      "query_prot GO:0007051 0.01115 spindle organization\n",
      "query_prot GO:0022008 0.01110 neurogenesis\n",
      "query_prot GO:0010038 0.01091 response to metal ion\n",
      "query_prot GO:0031668 0.01083 cellular response to extracellular stimulus\n",
      "query_prot GO:0006955 0.01080 immune response\n",
      "query_prot GO:0016567 0.01067 protein ubiquitination\n",
      "query_prot GO:0006302 0.01067 double-strand break repair\n",
      "query_prot GO:0009991 0.01058 response to extracellular stimulus\n",
      "query_prot GO:0044087 0.01055 regulation of cellular component biogenesis\n",
      "query_prot GO:0042981 0.01052 regulation of apoptotic process\n",
      "query_prot GO:0042330 0.01051 taxis\n",
      "query_prot GO:1901701 0.01044 cellular response to oxygen-containing compound\n",
      "query_prot GO:0050801 0.01040 monoatomic ion homeostasis\n",
      "query_prot GO:0006260 0.01027 DNA replication\n",
      "query_prot GO:0010629 0.01020 negative regulation of gene expression\n",
      "query_prot GO:0040011 0.01018 locomotion\n",
      "query_prot GO:0046649 0.01018 lymphocyte activation\n",
      "query_prot GO:0051276 0.01007 chromosome organization\n",
      "query_prot GO:0048736 0.01004 appendage development\n",
      "query_prot GO:0061061 0.01000 muscle structure development\n",
      "query_prot GO:0009267 0.00998 cellular response to starvation\n",
      "query_prot GO:0006605 0.00990 protein targeting\n",
      "query_prot GO:0051668 0.00984 localization within membrane\n",
      "query_prot GO:0051240 0.00980 positive regulation of multicellular organismal process\n",
      "query_prot GO:0044770 0.00979 cell cycle phase transition\n",
      "query_prot GO:0006403 0.00976 RNA localization\n",
      "query_prot GO:0033993 0.00975 response to lipid\n",
      "query_prot GO:0009411 0.00973 response to UV\n",
      "query_prot GO:0023051 0.00972 regulation of signaling\n",
      "query_prot GO:0000902 0.00972 cell morphogenesis\n",
      "query_prot GO:0019827 0.00963 stem cell population maintenance\n",
      "query_prot GO:0006979 0.00961 response to oxidative stress\n",
      "query_prot GO:0043067 0.00961 regulation of programmed cell death\n",
      "query_prot GO:0022412 0.00958 cellular process involved in reproduction in multicellular organism\n",
      "query_prot GO:0043161 0.00954 proteasome-mediated ubiquitin-dependent protein catabolic process\n",
      "query_prot GO:0120031 0.00953 plasma membrane bounded cell projection assembly\n",
      "query_prot GO:0006886 0.00942 intracellular protein transport\n",
      "query_prot GO:1902903 0.00940 regulation of supramolecular fiber organization\n",
      "query_prot GO:0044283 0.00935 small molecule biosynthetic process\n",
      "query_prot GO:0035050 0.00920 embryonic heart tube development\n",
      "query_prot GO:0006511 0.00906 ubiquitin-dependent protein catabolic process\n",
      "query_prot GO:0048871 0.00905 multicellular organismal-level homeostasis\n",
      "query_prot GO:0031647 0.00902 regulation of protein stability\n",
      "query_prot GO:0048863 0.00901 stem cell differentiation\n",
      "query_prot GO:0032870 0.00898 cellular response to hormone stimulus\n",
      "query_prot GO:0060041 0.00895 retina development in camera-type eye\n",
      "query_prot GO:0080135 0.00895 regulation of cellular response to stress\n",
      "query_prot GO:0006417 0.00886 regulation of translation\n",
      "query_prot GO:0043254 0.00880 regulation of protein-containing complex assembly\n",
      "query_prot GO:0019220 0.00876 regulation of phosphate metabolic process\n",
      "query_prot GO:0072359 0.00875 circulatory system development\n",
      "query_prot GO:0032879 0.00873 regulation of localization\n",
      "query_prot GO:0006970 0.00870 response to osmotic stress\n",
      "query_prot GO:0006091 0.00866 generation of precursor metabolites and energy\n",
      "query_prot GO:0070482 0.00849 response to oxygen levels\n",
      "query_prot GO:0006897 0.00841 endocytosis\n",
      "query_prot GO:0031327 0.00838 negative regulation of cellular biosynthetic process\n",
      "query_prot GO:0006873 0.00837 intracellular monoatomic ion homeostasis\n",
      "query_prot GO:0104004 0.00832 cellular response to environmental stimulus\n",
      "query_prot GO:0051254 0.00830 positive regulation of RNA metabolic process\n",
      "query_prot GO:0045595 0.00826 regulation of cell differentiation\n",
      "query_prot GO:0006325 0.00821 chromatin organization\n",
      "query_prot GO:0007346 0.00810 regulation of mitotic cell cycle\n",
      "query_prot GO:0031669 0.00809 cellular response to nutrient levels\n",
      "query_prot GO:0000377 0.00807 RNA splicing, via transesterification reactions with bulged adenosine as nucleophile\n",
      "query_prot GO:0051259 0.00805 protein complex oligomerization\n",
      "query_prot GO:0040007 0.00800 growth\n",
      "query_prot GO:0006338 0.00797 chromatin remodeling\n",
      "query_prot GO:0051052 0.00797 regulation of DNA metabolic process\n",
      "query_prot GO:0007127 0.00791 meiosis I\n",
      "query_prot GO:0030029 0.00784 actin filament-based process\n",
      "query_prot GO:0071496 0.00775 cellular response to external stimulus\n",
      "query_prot GO:0071363 0.00773 cellular response to growth factor stimulus\n",
      "query_prot GO:0071453 0.00759 cellular response to oxygen levels\n",
      "query_prot GO:0043207 0.00758 response to external biotic stimulus\n",
      "query_prot GO:0051174 0.00751 regulation of phosphorus metabolic process\n",
      "query_prot GO:0010927 0.00748 cellular component assembly involved in morphogenesis\n",
      "query_prot GO:0051604 0.00748 protein maturation\n",
      "query_prot GO:0055085 0.00746 transmembrane transport\n",
      "query_prot GO:0120036 0.00741 plasma membrane bounded cell projection organization\n",
      "query_prot GO:0034220 0.00741 monoatomic ion transmembrane transport\n",
      "query_prot GO:0071214 0.00740 cellular response to abiotic stimulus\n",
      "query_prot GO:0043065 0.00730 positive regulation of apoptotic process\n",
      "query_prot GO:0045926 0.00727 negative regulation of growth\n",
      "query_prot GO:0009408 0.00725 response to heat\n",
      "query_prot GO:0015931 0.00725 nucleobase-containing compound transport\n",
      "query_prot GO:0007286 0.00724 spermatid development\n",
      "query_prot GO:1902532 0.00717 negative regulation of intracellular signal transduction\n",
      "query_prot GO:0048638 0.00715 regulation of developmental growth\n",
      "query_prot GO:0040012 0.00713 regulation of locomotion\n",
      "query_prot GO:1901698 0.00707 response to nitrogen compound\n",
      "query_prot GO:1901137 0.00707 carbohydrate derivative biosynthetic process\n",
      "query_prot GO:0048729 0.00703 tissue morphogenesis\n",
      "query_prot GO:0043009 0.00700 chordate embryonic development\n",
      "query_prot GO:0010608 0.00694 post-transcriptional regulation of gene expression\n",
      "query_prot GO:0006468 0.00691 protein phosphorylation\n",
      "query_prot GO:0009895 0.00684 negative regulation of catabolic process\n",
      "query_prot GO:0098609 0.00677 cell-cell adhesion\n",
      "query_prot GO:0048699 0.00675 generation of neurons\n",
      "query_prot GO:0030036 0.00672 actin cytoskeleton organization\n",
      "query_prot GO:0051093 0.00670 negative regulation of developmental process\n",
      "query_prot GO:0051603 0.00666 proteolysis involved in protein catabolic process\n",
      "query_prot GO:0001775 0.00665 cell activation\n",
      "query_prot GO:0034470 0.00658 ncRNA processing\n",
      "query_prot GO:0009410 0.00653 response to xenobiotic stimulus\n",
      "query_prot GO:0030030 0.00652 cell projection organization\n",
      "query_prot GO:0045814 0.00642 negative regulation of gene expression, epigenetic\n",
      "query_prot GO:0048666 0.00642 neuron development\n",
      "query_prot GO:0033043 0.00639 regulation of organelle organization\n",
      "query_prot GO:0007368 0.00638 determination of left/right symmetry\n",
      "query_prot GO:0007267 0.00633 cell-cell signaling\n",
      "query_prot GO:0071900 0.00631 regulation of protein serine/threonine kinase activity\n",
      "query_prot GO:0048568 0.00630 embryonic organ development\n",
      "query_prot GO:0031330 0.00623 negative regulation of cellular catabolic process\n",
      "query_prot GO:1902679 0.00621 negative regulation of RNA biosynthetic process\n",
      "query_prot GO:0090596 0.00620 sensory organ morphogenesis\n",
      "query_prot GO:0045333 0.00618 cellular respiration\n",
      "query_prot GO:0043066 0.00614 negative regulation of apoptotic process\n",
      "query_prot GO:0050790 0.00613 regulation of catalytic activity\n",
      "query_prot GO:0009798 0.00612 axis specification\n",
      "query_prot GO:0022411 0.00609 cellular component disassembly\n",
      "query_prot GO:0001666 0.00607 response to hypoxia\n",
      "query_prot GO:0045859 0.00605 regulation of protein kinase activity\n",
      "query_prot GO:0048762 0.00605 mesenchymal cell differentiation\n",
      "query_prot GO:0071466 0.00600 cellular response to xenobiotic stimulus\n",
      "query_prot GO:0007163 0.00592 establishment or maintenance of cell polarity\n",
      "query_prot GO:0043010 0.00590 camera-type eye development\n",
      "query_prot GO:0007005 0.00585 mitochondrion organization\n",
      "query_prot GO:0055001 0.00584 muscle cell development\n",
      "query_prot GO:0048639 0.00582 positive regulation of developmental growth\n",
      "query_prot GO:0071417 0.00580 cellular response to organonitrogen compound\n",
      "query_prot GO:0071241 0.00576 cellular response to inorganic substance\n",
      "query_prot GO:0034248 0.00574 regulation of amide metabolic process\n",
      "query_prot GO:0060972 0.00572 left/right pattern formation\n",
      "query_prot GO:0031400 0.00570 negative regulation of protein modification process\n",
      "query_prot GO:0007417 0.00568 central nervous system development\n",
      "query_prot GO:0051338 0.00568 regulation of transferase activity\n",
      "query_prot GO:0044419 0.00568 biological process involved in interspecies interaction between organisms\n",
      "query_prot GO:0019637 0.00567 organophosphate metabolic process\n",
      "query_prot GO:0097305 0.00566 response to alcohol\n",
      "query_prot GO:0009266 0.00561 response to temperature stimulus\n",
      "query_prot GO:0071495 0.00561 cellular response to endogenous stimulus\n",
      "query_prot GO:0080134 0.00555 regulation of response to stress\n",
      "query_prot GO:0000122 0.00553 negative regulation of transcription by RNA polymerase II\n",
      "query_prot GO:0031399 0.00553 regulation of protein modification process\n",
      "query_prot GO:0006364 0.00550 rRNA processing\n",
      "query_prot GO:0045229 0.00547 external encapsulating structure organization\n",
      "query_prot GO:0008380 0.00547 RNA splicing\n",
      "query_prot GO:0040008 0.00547 regulation of growth\n",
      "query_prot GO:0021537 0.00545 telencephalon development\n",
      "query_prot GO:0010646 0.00540 regulation of cell communication\n",
      "query_prot GO:0043068 0.00539 positive regulation of programmed cell death\n",
      "query_prot GO:0032102 0.00537 negative regulation of response to external stimulus\n",
      "query_prot GO:0051241 0.00526 negative regulation of multicellular organismal process\n",
      "query_prot GO:0032940 0.00522 secretion by cell\n",
      "query_prot GO:0051094 0.00522 positive regulation of developmental process\n",
      "query_prot GO:0055074 0.00521 calcium ion homeostasis\n",
      "query_prot GO:0001894 0.00519 tissue homeostasis\n",
      "query_prot GO:0030155 0.00514 regulation of cell adhesion\n",
      "query_prot GO:0043632 0.00512 modification-dependent macromolecule catabolic process\n",
      "query_prot GO:0050678 0.00507 regulation of epithelial cell proliferation\n",
      "query_prot GO:0007369 0.00507 gastrulation\n",
      "query_prot GO:0048592 0.00507 eye morphogenesis\n",
      "query_prot GO:0035239 0.00504 tube morphogenesis\n",
      "query_prot GO:0008033 0.00504 tRNA processing\n",
      "query_prot GO:0050673 0.00501 epithelial cell proliferation\n",
      "query_prot GO:0007610 0.00501 behavior\n",
      "query_prot GO:0009416 0.00497 response to light stimulus\n",
      "query_prot GO:0032101 0.00493 regulation of response to external stimulus\n",
      "query_prot GO:0002521 0.00493 leukocyte differentiation\n",
      "query_prot GO:0009968 0.00492 negative regulation of signal transduction\n",
      "query_prot GO:2000243 0.00492 positive regulation of reproductive process\n",
      "query_prot GO:0043410 0.00491 positive regulation of MAPK cascade\n",
      "query_prot GO:0009886 0.00491 post-embryonic animal morphogenesis\n",
      "query_prot GO:0040029 0.00484 epigenetic regulation of gene expression\n",
      "query_prot GO:0044255 0.00484 cellular lipid metabolic process\n",
      "query_prot GO:0031347 0.00482 regulation of defense response\n",
      "query_prot GO:0045927 0.00479 positive regulation of growth\n",
      "query_prot GO:1901135 0.00478 carbohydrate derivative metabolic process\n",
      "query_prot GO:0010647 0.00476 positive regulation of cell communication\n",
      "query_prot GO:0009259 0.00473 ribonucleotide metabolic process\n",
      "query_prot GO:0045935 0.00472 positive regulation of nucleobase-containing compound metabolic process\n",
      "query_prot GO:0016310 0.00471 phosphorylation\n",
      "query_prot GO:0006812 0.00471 monoatomic cation transport\n",
      "query_prot GO:0048593 0.00471 camera-type eye morphogenesis\n",
      "query_prot GO:0008015 0.00469 blood circulation\n",
      "query_prot GO:1901652 0.00469 response to peptide\n",
      "query_prot GO:0045934 0.00469 negative regulation of nucleobase-containing compound metabolic process\n",
      "query_prot GO:0006261 0.00468 DNA-templated DNA replication\n",
      "query_prot GO:0048367 0.00464 shoot system development\n",
      "query_prot GO:0007423 0.00463 sensory organ development\n",
      "query_prot GO:2000026 0.00463 regulation of multicellular organismal development\n",
      "query_prot GO:0051253 0.00462 negative regulation of RNA metabolic process\n",
      "query_prot GO:0048580 0.00461 regulation of post-embryonic development\n",
      "query_prot GO:0000070 0.00459 mitotic sister chromatid segregation\n",
      "query_prot GO:0000302 0.00458 response to reactive oxygen species\n",
      "query_prot GO:0098660 0.00452 inorganic ion transmembrane transport\n",
      "query_prot GO:0019725 0.00448 cellular homeostasis\n",
      "query_prot GO:0045892 0.00446 negative regulation of DNA-templated transcription\n",
      "query_prot GO:0032970 0.00445 regulation of actin filament-based process\n",
      "query_prot GO:0098727 0.00444 maintenance of cell number\n",
      "query_prot GO:0010639 0.00444 negative regulation of organelle organization\n",
      "query_prot GO:0043269 0.00440 regulation of monoatomic ion transport\n",
      "query_prot GO:0098662 0.00432 inorganic cation transmembrane transport\n",
      "query_prot GO:0017148 0.00431 negative regulation of translation\n",
      "query_prot GO:0001817 0.00430 regulation of cytokine production\n",
      "query_prot GO:0009636 0.00427 response to toxic substance\n",
      "query_prot GO:0006163 0.00425 purine nucleotide metabolic process\n",
      "query_prot GO:1901699 0.00424 cellular response to nitrogen compound\n",
      "query_prot GO:0009165 0.00419 nucleotide biosynthetic process\n",
      "query_prot GO:0060284 0.00418 regulation of cell development\n",
      "query_prot GO:1901990 0.00414 regulation of mitotic cell cycle phase transition\n",
      "query_prot GO:0046395 0.00413 carboxylic acid catabolic process\n",
      "query_prot GO:0030182 0.00413 neuron differentiation\n",
      "query_prot GO:0048477 0.00413 oogenesis\n",
      "query_prot GO:0019439 0.00410 aromatic compound catabolic process\n",
      "query_prot GO:0009117 0.00405 nucleotide metabolic process\n",
      "query_prot GO:0006954 0.00399 inflammatory response\n",
      "query_prot GO:0022604 0.00399 regulation of cell morphogenesis\n",
      "query_prot GO:0051701 0.00398 biological process involved in interaction with host\n",
      "query_prot GO:0032787 0.00397 monocarboxylic acid metabolic process\n",
      "query_prot GO:0034976 0.00397 response to endoplasmic reticulum stress\n",
      "query_prot GO:0009617 0.00397 response to bacterium\n",
      "query_prot GO:0048515 0.00395 spermatid differentiation\n",
      "query_prot GO:0030855 0.00391 epithelial cell differentiation\n",
      "query_prot GO:0051051 0.00390 negative regulation of transport\n",
      "query_prot GO:0070085 0.00390 glycosylation\n",
      "query_prot GO:0009101 0.00389 glycoprotein biosynthetic process\n",
      "query_prot GO:0016477 0.00388 cell migration\n",
      "query_prot GO:0030001 0.00387 metal ion transport\n",
      "query_prot GO:0002009 0.00386 morphogenesis of an epithelium\n",
      "query_prot GO:0016053 0.00386 organic acid biosynthetic process\n",
      "query_prot GO:0051493 0.00386 regulation of cytoskeleton organization\n",
      "query_prot GO:0061458 0.00385 reproductive system development\n",
      "query_prot GO:0006082 0.00382 organic acid metabolic process\n",
      "query_prot GO:1903311 0.00382 regulation of mRNA metabolic process\n",
      "query_prot GO:0050808 0.00382 synapse organization\n",
      "query_prot GO:0045937 0.00381 positive regulation of phosphate metabolic process\n",
      "query_prot GO:1901361 0.00381 organic cyclic compound catabolic process\n",
      "query_prot GO:0055086 0.00379 nucleobase-containing small molecule metabolic process\n",
      "query_prot GO:0048545 0.00379 response to steroid hormone\n",
      "query_prot GO:0061024 0.00379 membrane organization\n",
      "query_prot GO:0045787 0.00377 positive regulation of cell cycle\n",
      "query_prot GO:0022407 0.00375 regulation of cell-cell adhesion\n",
      "query_prot GO:0007018 0.00374 microtubule-based movement\n",
      "query_prot GO:0045930 0.00371 negative regulation of mitotic cell cycle\n",
      "query_prot GO:2000145 0.00370 regulation of cell motility\n",
      "query_prot GO:0051049 0.00369 regulation of transport\n",
      "query_prot GO:0001933 0.00369 negative regulation of protein phosphorylation\n",
      "query_prot GO:0098655 0.00367 monoatomic cation transmembrane transport\n",
      "query_prot GO:1901987 0.00365 regulation of cell cycle phase transition\n",
      "query_prot GO:0048878 0.00365 chemical homeostasis\n",
      "query_prot GO:0042594 0.00363 response to starvation\n",
      "query_prot GO:0044003 0.00361 modulation by symbiont of host process\n",
      "query_prot GO:0043087 0.00357 regulation of GTPase activity\n",
      "query_prot GO:0032989 0.00352 cellular component morphogenesis\n",
      "query_prot GO:0051640 0.00351 organelle localization\n",
      "query_prot GO:0050727 0.00350 regulation of inflammatory response\n",
      "query_prot GO:0071456 0.00350 cellular response to hypoxia\n",
      "query_prot GO:0034097 0.00350 response to cytokine\n",
      "query_prot GO:1902533 0.00349 positive regulation of intracellular signal transduction\n",
      "query_prot GO:0044706 0.00344 multi-multicellular organism process\n",
      "query_prot GO:0048193 0.00341 Golgi vesicle transport\n",
      "query_prot GO:1903131 0.00341 mononuclear cell differentiation\n",
      "query_prot GO:0048589 0.00340 developmental growth\n",
      "query_prot GO:1902074 0.00338 response to salt\n",
      "query_prot GO:1901653 0.00338 cellular response to peptide\n",
      "query_prot GO:0042063 0.00337 gliogenesis\n",
      "query_prot GO:0070647 0.00336 protein modification by small protein conjugation or removal\n",
      "query_prot GO:1904888 0.00334 cranial skeletal system development\n",
      "query_prot GO:0033044 0.00333 regulation of chromosome organization\n",
      "query_prot GO:0032990 0.00331 cell part morphogenesis\n",
      "query_prot GO:1902531 0.00331 regulation of intracellular signal transduction\n",
      "query_prot GO:0055080 0.00331 monoatomic cation homeostasis\n",
      "query_prot GO:0044772 0.00329 mitotic cell cycle phase transition\n",
      "query_prot GO:0072594 0.00328 establishment of protein localization to organelle\n",
      "query_prot GO:0016485 0.00327 protein processing\n",
      "query_prot GO:0007167 0.00326 enzyme-linked receptor protein signaling pathway\n",
      "query_prot GO:0031032 0.00325 actomyosin structure organization\n",
      "query_prot GO:0046903 0.00324 secretion\n",
      "query_prot GO:0032259 0.00324 methylation\n",
      "query_prot GO:0043408 0.00321 regulation of MAPK cascade\n",
      "query_prot GO:0001934 0.00319 positive regulation of protein phosphorylation\n",
      "query_prot GO:0036293 0.00318 response to decreased oxygen levels\n",
      "query_prot GO:0018193 0.00315 peptidyl-amino acid modification\n",
      "query_prot GO:0009737 0.00313 response to abscisic acid\n",
      "query_prot GO:0001701 0.00312 in utero embryonic development\n",
      "query_prot GO:2000146 0.00306 negative regulation of cell motility\n",
      "query_prot GO:0001745 0.00305 compound eye morphogenesis\n",
      "query_prot GO:0008340 0.00302 determination of adult lifespan\n",
      "query_prot GO:0016032 0.00302 viral process\n",
      "query_prot GO:0016049 0.00301 cell growth\n",
      "query_prot GO:0010562 0.00298 positive regulation of phosphorus metabolic process\n",
      "query_prot GO:0050776 0.00297 regulation of immune response\n",
      "query_prot GO:0030003 0.00297 intracellular monoatomic cation homeostasis\n",
      "query_prot GO:0030335 0.00296 positive regulation of cell migration\n",
      "query_prot GO:0051347 0.00296 positive regulation of transferase activity\n",
      "query_prot GO:0006109 0.00295 regulation of carbohydrate metabolic process\n",
      "query_prot GO:0007178 0.00295 transmembrane receptor protein serine/threonine kinase signaling pathway\n",
      "query_prot GO:0055082 0.00295 intracellular chemical homeostasis\n",
      "query_prot GO:0075136 0.00294 response to host\n",
      "query_prot GO:0050832 0.00293 defense response to fungus\n",
      "query_prot GO:0006914 0.00291 autophagy\n",
      "query_prot GO:0019752 0.00290 carboxylic acid metabolic process\n",
      "query_prot GO:0043434 0.00288 response to peptide hormone\n",
      "query_prot GO:0060562 0.00285 epithelial tube morphogenesis\n",
      "query_prot GO:1901888 0.00285 regulation of cell junction assembly\n",
      "query_prot GO:0032535 0.00283 regulation of cellular component size\n",
      "query_prot GO:0048839 0.00280 inner ear development\n",
      "query_prot GO:0007560 0.00278 imaginal disc morphogenesis\n",
      "query_prot GO:0090257 0.00277 regulation of muscle system process\n",
      "query_prot GO:0016052 0.00275 carbohydrate catabolic process\n",
      "query_prot GO:0030162 0.00273 regulation of proteolysis\n",
      "query_prot GO:0045596 0.00273 negative regulation of cell differentiation\n",
      "query_prot GO:0051216 0.00272 cartilage development\n",
      "query_prot GO:0019693 0.00271 ribose phosphate metabolic process\n",
      "query_prot GO:0051130 0.00267 positive regulation of cellular component organization\n",
      "query_prot GO:0006816 0.00265 calcium ion transport\n",
      "query_prot GO:1902075 0.00264 cellular response to salt\n",
      "query_prot GO:0033674 0.00263 positive regulation of kinase activity\n",
      "query_prot GO:0006997 0.00263 nucleus organization\n",
      "query_prot GO:0071345 0.00262 cellular response to cytokine stimulus\n",
      "query_prot GO:0006486 0.00261 protein glycosylation\n",
      "query_prot GO:0048563 0.00258 post-embryonic animal organ morphogenesis\n",
      "query_prot GO:0098771 0.00257 inorganic ion homeostasis\n",
      "query_prot GO:0034329 0.00256 cell junction assembly\n",
      "query_prot GO:0034330 0.00255 cell junction organization\n",
      "query_prot GO:0006753 0.00255 nucleoside phosphate metabolic process\n",
      "query_prot GO:0032956 0.00255 regulation of actin cytoskeleton organization\n",
      "query_prot GO:0050804 0.00254 modulation of chemical synaptic transmission\n",
      "query_prot GO:0001754 0.00254 eye photoreceptor cell differentiation\n",
      "query_prot GO:0014070 0.00254 response to organic cyclic compound\n",
      "query_prot GO:0010876 0.00253 lipid localization\n",
      "query_prot GO:0006629 0.00253 lipid metabolic process\n",
      "query_prot GO:0008284 0.00253 positive regulation of cell population proliferation\n",
      "query_prot GO:0048562 0.00252 embryonic organ morphogenesis\n",
      "query_prot GO:0007292 0.00252 female gamete generation\n",
      "query_prot GO:0002699 0.00251 positive regulation of immune effector process\n",
      "query_prot GO:0001503 0.00250 ossification\n",
      "query_prot GO:1902850 0.00248 microtubule cytoskeleton organization involved in mitosis\n",
      "query_prot GO:0023057 0.00247 negative regulation of signaling\n",
      "query_prot GO:0001944 0.00246 vasculature development\n",
      "query_prot GO:0051336 0.00246 regulation of hydrolase activity\n",
      "query_prot GO:0031348 0.00241 negative regulation of defense response\n",
      "query_prot GO:0000910 0.00240 cytokinesis\n",
      "query_prot GO:1901654 0.00239 response to ketone\n",
      "query_prot GO:0071396 0.00238 cellular response to lipid\n",
      "query_prot GO:0043436 0.00238 oxoacid metabolic process\n",
      "query_prot GO:0006935 0.00237 chemotaxis\n",
      "query_prot GO:1905114 0.00235 cell surface receptor signaling pathway involved in cell-cell signaling\n",
      "query_prot GO:1990778 0.00234 protein localization to cell periphery\n",
      "query_prot GO:0021700 0.00232 developmental maturation\n",
      "query_prot GO:0072522 0.00230 purine-containing compound biosynthetic process\n",
      "query_prot GO:0048584 0.00230 positive regulation of response to stimulus\n",
      "query_prot GO:0055002 0.00229 striated muscle cell development\n",
      "query_prot GO:0042692 0.00228 muscle cell differentiation\n",
      "query_prot GO:0051223 0.00228 regulation of protein transport\n",
      "query_prot GO:0036294 0.00228 cellular response to decreased oxygen levels\n",
      "query_prot GO:0060560 0.00227 developmental growth involved in morphogenesis\n",
      "query_prot GO:0044272 0.00227 sulfur compound biosynthetic process\n",
      "query_prot GO:2000147 0.00227 positive regulation of cell motility\n",
      "query_prot GO:0031346 0.00227 positive regulation of cell projection organization\n",
      "query_prot GO:0043414 0.00225 macromolecule methylation\n",
      "query_prot GO:0007015 0.00225 actin filament organization\n",
      "query_prot GO:0052548 0.00224 regulation of endopeptidase activity\n",
      "query_prot GO:0032103 0.00224 positive regulation of response to external stimulus\n",
      "query_prot GO:0003012 0.00224 muscle system process\n",
      "query_prot GO:0060537 0.00223 muscle tissue development\n",
      "query_prot GO:0071478 0.00223 cellular response to radiation\n",
      "query_prot GO:0099402 0.00222 plant organ development\n",
      "query_prot GO:0048732 0.00222 gland development\n",
      "query_prot GO:0009150 0.00221 purine ribonucleotide metabolic process\n",
      "query_prot GO:0034249 0.00221 negative regulation of amide metabolic process\n",
      "query_prot GO:0048704 0.00221 embryonic skeletal system morphogenesis\n",
      "query_prot GO:0042546 0.00217 cell wall biogenesis\n",
      "query_prot GO:0002697 0.00217 regulation of immune effector process\n",
      "query_prot GO:0002165 0.00216 instar larval or pupal development\n",
      "query_prot GO:0071695 0.00216 anatomical structure maturation\n",
      "query_prot GO:0051054 0.00216 positive regulation of DNA metabolic process\n",
      "query_prot GO:0035107 0.00215 appendage morphogenesis\n",
      "query_prot GO:0045321 0.00215 leukocyte activation\n",
      "query_prot GO:0009880 0.00213 embryonic pattern specification\n",
      "query_prot GO:0002682 0.00212 regulation of immune system process\n",
      "query_prot GO:0009896 0.00212 positive regulation of catabolic process\n",
      "query_prot GO:0072521 0.00212 purine-containing compound metabolic process\n",
      "query_prot GO:0022622 0.00212 root system development\n",
      "query_prot GO:0006399 0.00211 tRNA metabolic process\n",
      "query_prot GO:0052200 0.00211 response to host defenses\n",
      "query_prot GO:0009620 0.00210 response to fungus\n",
      "query_prot GO:0070848 0.00207 response to growth factor\n",
      "query_prot GO:0050878 0.00206 regulation of body fluid levels\n",
      "query_prot GO:0010648 0.00204 negative regulation of cell communication\n",
      "query_prot GO:0198738 0.00203 cell-cell signaling by wnt\n",
      "query_prot GO:0007600 0.00202 sensory perception\n",
      "query_prot GO:0061640 0.00202 cytoskeleton-dependent cytokinesis\n",
      "query_prot GO:0044089 0.00202 positive regulation of cellular component biogenesis\n",
      "query_prot GO:0007162 0.00201 negative regulation of cell adhesion\n",
      "query_prot GO:0098916 0.00200 anterograde trans-synaptic signaling\n",
      "query_prot GO:0030099 0.00198 myeloid cell differentiation\n",
      "query_prot GO:0007626 0.00197 locomotory behavior\n",
      "query_prot GO:0048588 0.00197 developmental cell growth\n",
      "query_prot GO:0099537 0.00196 trans-synaptic signaling\n",
      "query_prot GO:0030447 0.00194 filamentous growth\n",
      "query_prot GO:1901214 0.00194 regulation of neuron death\n",
      "query_prot GO:0007472 0.00193 wing disc morphogenesis\n",
      "query_prot GO:0009100 0.00193 glycoprotein metabolic process\n",
      "query_prot GO:0031344 0.00193 regulation of cell projection organization\n",
      "query_prot GO:0048569 0.00192 post-embryonic animal organ development\n",
      "query_prot GO:0046394 0.00191 carboxylic acid biosynthetic process\n",
      "query_prot GO:0048872 0.00191 homeostasis of number of cells\n",
      "query_prot GO:0030902 0.00190 hindbrain development\n",
      "query_prot GO:0044403 0.00188 biological process involved in symbiotic interaction\n",
      "query_prot GO:0007552 0.00186 metamorphosis\n",
      "query_prot GO:0044282 0.00186 small molecule catabolic process\n",
      "query_prot GO:0035821 0.00186 modulation of process of another organism\n",
      "query_prot GO:0000165 0.00183 MAPK cascade\n",
      "query_prot GO:0050803 0.00183 regulation of synapse structure or activity\n",
      "query_prot GO:0061919 0.00181 process utilizing autophagic mechanism\n",
      "query_prot GO:0045785 0.00179 positive regulation of cell adhesion\n",
      "query_prot GO:0002684 0.00174 positive regulation of immune system process\n",
      "query_prot GO:0043069 0.00172 negative regulation of programmed cell death\n",
      "query_prot GO:0007411 0.00172 axon guidance\n",
      "query_prot GO:0043549 0.00170 regulation of kinase activity\n",
      "query_prot GO:0007519 0.00170 skeletal muscle tissue development\n",
      "query_prot GO:0008610 0.00170 lipid biosynthetic process\n",
      "query_prot GO:0016054 0.00169 organic acid catabolic process\n",
      "query_prot GO:0045597 0.00168 positive regulation of cell differentiation\n",
      "query_prot GO:0070997 0.00168 neuron death\n",
      "query_prot GO:0071554 0.00167 cell wall organization or biogenesis\n",
      "query_prot GO:0050778 0.00167 positive regulation of immune response\n",
      "query_prot GO:0006644 0.00166 phospholipid metabolic process\n",
      "query_prot GO:0060249 0.00165 anatomical structure homeostasis\n",
      "query_prot GO:0045786 0.00164 negative regulation of cell cycle\n",
      "query_prot GO:0071248 0.00163 cellular response to metal ion\n",
      "query_prot GO:0043413 0.00163 macromolecule glycosylation\n",
      "query_prot GO:0050767 0.00161 regulation of neurogenesis\n",
      "query_prot GO:0010817 0.00159 regulation of hormone levels\n",
      "query_prot GO:0071407 0.00159 cellular response to organic cyclic compound\n",
      "query_prot GO:0072330 0.00158 monocarboxylic acid biosynthetic process\n",
      "query_prot GO:0043583 0.00158 ear development\n",
      "query_prot GO:0110053 0.00158 regulation of actin filament organization\n",
      "query_prot GO:0010959 0.00157 regulation of metal ion transport\n",
      "query_prot GO:0061053 0.00157 somite development\n",
      "query_prot GO:0010035 0.00156 response to inorganic substance\n",
      "query_prot GO:0061564 0.00156 axon development\n",
      "query_prot GO:0035220 0.00155 wing disc development\n",
      "query_prot GO:0034762 0.00155 regulation of transmembrane transport\n",
      "query_prot GO:0007444 0.00154 imaginal disc development\n",
      "query_prot GO:0048827 0.00154 phyllome development\n",
      "query_prot GO:0022603 0.00152 regulation of anatomical structure morphogenesis\n",
      "query_prot GO:0045132 0.00150 meiotic chromosome segregation\n",
      "query_prot GO:0045471 0.00149 response to ethanol\n",
      "query_prot GO:0007548 0.00149 sex differentiation\n",
      "query_prot GO:0051606 0.00146 detection of stimulus\n",
      "query_prot GO:0035120 0.00146 post-embryonic appendage morphogenesis\n",
      "query_prot GO:0016236 0.00146 macroautophagy\n",
      "query_prot GO:0009651 0.00146 response to salt stress\n",
      "query_prot GO:0032880 0.00145 regulation of protein localization\n",
      "query_prot GO:0006869 0.00144 lipid transport\n",
      "query_prot GO:0051090 0.00144 regulation of DNA-binding transcription factor activity\n",
      "query_prot GO:0060627 0.00143 regulation of vesicle-mediated transport\n",
      "query_prot GO:0055123 0.00143 digestive system development\n",
      "query_prot GO:0009952 0.00143 anterior/posterior pattern specification\n",
      "query_prot GO:1901215 0.00142 negative regulation of neuron death\n",
      "query_prot GO:0008406 0.00142 gonad development\n",
      "query_prot GO:0140352 0.00142 export from cell\n",
      "query_prot GO:0006164 0.00140 purine nucleotide biosynthetic process\n",
      "query_prot GO:0045165 0.00140 cell fate commitment\n",
      "query_prot GO:0009611 0.00140 response to wounding\n",
      "query_prot GO:0009894 0.00138 regulation of catabolic process\n",
      "query_prot GO:0070201 0.00138 regulation of establishment of protein localization\n",
      "query_prot GO:0008361 0.00137 regulation of cell size\n",
      "query_prot GO:0015711 0.00137 organic anion transport\n",
      "query_prot GO:0010720 0.00137 positive regulation of cell development\n",
      "query_prot GO:0046700 0.00136 heterocycle catabolic process\n",
      "query_prot GO:0042110 0.00136 T cell activation\n",
      "query_prot GO:0031331 0.00135 positive regulation of cellular catabolic process\n",
      "query_prot GO:0051607 0.00133 defense response to virus\n",
      "query_prot GO:0010563 0.00132 negative regulation of phosphorus metabolic process\n",
      "query_prot GO:0003013 0.00131 circulatory system process\n",
      "query_prot GO:0048706 0.00131 embryonic skeletal system development\n",
      "query_prot GO:0045087 0.00129 innate immune response\n",
      "query_prot GO:0007584 0.00129 response to nutrient\n",
      "query_prot GO:0042391 0.00129 regulation of membrane potential\n",
      "query_prot GO:0050890 0.00127 cognition\n",
      "query_prot GO:0040013 0.00127 negative regulation of locomotion\n",
      "query_prot GO:0002237 0.00126 response to molecule of bacterial origin\n",
      "query_prot GO:0031175 0.00125 neuron projection development\n",
      "query_prot GO:0031329 0.00122 regulation of cellular catabolic process\n",
      "query_prot GO:0001525 0.00120 angiogenesis\n",
      "query_prot GO:0048229 0.00120 gametophyte development\n",
      "query_prot GO:0034599 0.00120 cellular response to oxidative stress\n",
      "query_prot GO:0044057 0.00120 regulation of system process\n",
      "query_prot GO:0021953 0.00119 central nervous system neuron differentiation\n",
      "query_prot GO:0031016 0.00118 pancreas development\n",
      "query_prot GO:0019058 0.00117 viral life cycle\n",
      "query_prot GO:0007186 0.00117 G protein-coupled receptor signaling pathway\n",
      "query_prot GO:0035282 0.00116 segmentation\n",
      "query_prot GO:0030239 0.00115 myofibril assembly\n",
      "query_prot GO:0099536 0.00115 synaptic signaling\n",
      "query_prot GO:0031349 0.00112 positive regulation of defense response\n",
      "query_prot GO:0048737 0.00112 imaginal disc-derived appendage development\n",
      "query_prot GO:0120039 0.00112 plasma membrane bounded cell projection morphogenesis\n",
      "query_prot GO:0048707 0.00111 instar larval or pupal morphogenesis\n",
      "query_prot GO:0031401 0.00109 positive regulation of protein modification process\n",
      "query_prot GO:0002164 0.00106 larval development\n",
      "query_prot GO:0019216 0.00106 regulation of lipid metabolic process\n",
      "query_prot GO:0000904 0.00106 cell morphogenesis involved in differentiation\n",
      "query_prot GO:0006520 0.00106 amino acid metabolic process\n",
      "query_prot GO:0045089 0.00105 positive regulation of innate immune response\n",
      "query_prot GO:0061982 0.00105 meiosis I cell cycle process\n",
      "query_prot GO:0052572 0.00104 response to host immune response\n",
      "query_prot GO:0001816 0.00104 cytokine production\n",
      "query_prot GO:0071692 0.00102 protein localization to extracellular region\n",
      "query_prot GO:0090407 0.00100 organophosphate biosynthetic process\n",
      "query_prot GO:0010975 0.00099 regulation of neuron projection development\n",
      "query_prot GO:0031099 0.00099 regeneration\n",
      "query_prot GO:0140546 0.00099 defense response to symbiont\n",
      "query_prot GO:0014706 0.00099 striated muscle tissue development\n",
      "query_prot GO:0002833 0.00099 positive regulation of response to biotic stimulus\n",
      "query_prot GO:0042060 0.00099 wound healing\n",
      "query_prot GO:0061008 0.00098 hepaticobiliary system development\n",
      "query_prot GO:0048749 0.00097 compound eye development\n",
      "query_prot GO:0090087 0.00097 regulation of peptide transport\n",
      "query_prot GO:0048364 0.00096 root development\n",
      "query_prot GO:0046883 0.00095 regulation of hormone secretion\n",
      "query_prot GO:0045936 0.00094 negative regulation of phosphate metabolic process\n",
      "query_prot GO:0034655 0.00093 nucleobase-containing compound catabolic process\n",
      "query_prot GO:0052173 0.00093 response to defenses of other organism\n",
      "query_prot GO:0016331 0.00092 morphogenesis of embryonic epithelium\n",
      "query_prot GO:0006633 0.00092 fatty acid biosynthetic process\n",
      "query_prot GO:0097190 0.00092 apoptotic signaling pathway\n",
      "query_prot GO:0051050 0.00092 positive regulation of transport\n",
      "query_prot GO:0045860 0.00092 positive regulation of protein kinase activity\n",
      "query_prot GO:0009409 0.00091 response to cold\n",
      "query_prot GO:0010948 0.00091 negative regulation of cell cycle process\n",
      "query_prot GO:0007033 0.00090 vacuole organization\n",
      "query_prot GO:0030198 0.00089 extracellular matrix organization\n",
      "query_prot GO:0030534 0.00089 adult behavior\n",
      "query_prot GO:1901605 0.00089 alpha-amino acid metabolic process\n",
      "query_prot GO:0051961 0.00089 negative regulation of nervous system development\n",
      "query_prot GO:0043062 0.00088 extracellular structure organization\n",
      "query_prot GO:0010721 0.00088 negative regulation of cell development\n",
      "query_prot GO:0099177 0.00087 regulation of trans-synaptic signaling\n",
      "query_prot GO:0009743 0.00087 response to carbohydrate\n",
      "query_prot GO:0001819 0.00086 positive regulation of cytokine production\n",
      "query_prot GO:0006909 0.00084 phagocytosis\n",
      "query_prot GO:0042327 0.00084 positive regulation of phosphorylation\n",
      "query_prot GO:0099111 0.00084 microtubule-based transport\n",
      "query_prot GO:0032386 0.00084 regulation of intracellular transport\n",
      "query_prot GO:0022898 0.00083 regulation of transmembrane transporter activity\n",
      "query_prot GO:0015849 0.00083 organic acid transport\n",
      "query_prot GO:0071375 0.00082 cellular response to peptide hormone stimulus\n",
      "query_prot GO:0001667 0.00082 ameboidal-type cell migration\n",
      "query_prot GO:0052547 0.00082 regulation of peptidase activity\n",
      "query_prot GO:0051146 0.00082 striated muscle cell differentiation\n",
      "query_prot GO:0002252 0.00081 immune effector process\n",
      "query_prot GO:0001889 0.00081 liver development\n",
      "query_prot GO:0032412 0.00081 regulation of monoatomic ion transmembrane transporter activity\n",
      "query_prot GO:0042742 0.00079 defense response to bacterium\n",
      "query_prot GO:0042176 0.00079 regulation of protein catabolic process\n",
      "query_prot GO:0051046 0.00078 regulation of secretion\n",
      "query_prot GO:1903530 0.00078 regulation of secretion by cell\n",
      "query_prot GO:0050865 0.00077 regulation of cell activation\n",
      "query_prot GO:0009755 0.00076 hormone-mediated signaling pathway\n",
      "query_prot GO:0023061 0.00076 signal release\n",
      "query_prot GO:0016311 0.00075 dephosphorylation\n",
      "query_prot GO:0001558 0.00075 regulation of cell growth\n",
      "query_prot GO:0040017 0.00075 positive regulation of locomotion\n",
      "query_prot GO:0060070 0.00075 canonical Wnt signaling pathway\n",
      "query_prot GO:0050708 0.00074 regulation of protein secretion\n",
      "query_prot GO:0051222 0.00073 positive regulation of protein transport\n",
      "query_prot GO:0001568 0.00072 blood vessel development\n",
      "query_prot GO:0003007 0.00072 heart morphogenesis\n",
      "query_prot GO:0030010 0.00072 establishment of cell polarity\n",
      "query_prot GO:0009914 0.00072 hormone transport\n",
      "query_prot GO:0010506 0.00071 regulation of autophagy\n",
      "query_prot GO:0006402 0.00071 mRNA catabolic process\n",
      "query_prot GO:0009451 0.00071 RNA modification\n",
      "query_prot GO:0044270 0.00071 cellular nitrogen compound catabolic process\n",
      "query_prot GO:0002253 0.00070 activation of immune response\n",
      "query_prot GO:0030100 0.00070 regulation of endocytosis\n",
      "query_prot GO:0051960 0.00070 regulation of nervous system development\n",
      "query_prot GO:0046890 0.00069 regulation of lipid biosynthetic process\n",
      "query_prot GO:0120035 0.00069 regulation of plasma membrane bounded cell projection organization\n",
      "query_prot GO:0060485 0.00068 mesenchyme development\n",
      "query_prot GO:0046777 0.00067 protein autophosphorylation\n",
      "query_prot GO:0045664 0.00067 regulation of neuron differentiation\n",
      "query_prot GO:1901988 0.00067 negative regulation of cell cycle phase transition\n",
      "query_prot GO:0030705 0.00066 cytoskeleton-dependent intracellular transport\n",
      "query_prot GO:1901293 0.00066 nucleoside phosphate biosynthetic process\n",
      "query_prot GO:0050829 0.00065 defense response to Gram-negative bacterium\n",
      "query_prot GO:0051345 0.00065 positive regulation of hydrolase activity\n",
      "query_prot GO:0007416 0.00065 synapse assembly\n",
      "query_prot GO:0008652 0.00064 amino acid biosynthetic process\n",
      "query_prot GO:0002831 0.00064 regulation of response to biotic stimulus\n",
      "query_prot GO:0042886 0.00064 amide transport\n",
      "query_prot GO:0016050 0.00064 vesicle organization\n",
      "query_prot GO:0006631 0.00061 fatty acid metabolic process\n",
      "query_prot GO:0030111 0.00060 regulation of Wnt signaling pathway\n",
      "query_prot GO:0007476 0.00060 imaginal disc-derived wing morphogenesis\n",
      "query_prot GO:0048738 0.00059 cardiac muscle tissue development\n",
      "query_prot GO:1904951 0.00058 positive regulation of establishment of protein localization\n",
      "query_prot GO:0048514 0.00058 blood vessel morphogenesis\n",
      "query_prot GO:0042326 0.00057 negative regulation of phosphorylation\n",
      "query_prot GO:0061013 0.00057 regulation of mRNA catabolic process\n",
      "query_prot GO:0050807 0.00056 regulation of synapse organization\n",
      "query_prot GO:0043473 0.00056 pigmentation\n",
      "query_prot GO:0006401 0.00055 RNA catabolic process\n",
      "query_prot GO:0048469 0.00055 cell maturation\n",
      "query_prot GO:1903532 0.00054 positive regulation of secretion by cell\n",
      "query_prot GO:0035114 0.00054 imaginal disc-derived appendage morphogenesis\n",
      "query_prot GO:0002683 0.00052 negative regulation of immune system process\n",
      "query_prot GO:0048812 0.00052 neuron projection morphogenesis\n",
      "query_prot GO:0002696 0.00050 positive regulation of leukocyte activation\n",
      "query_prot GO:0048858 0.00049 cell projection morphogenesis\n",
      "query_prot GO:0008544 0.00049 epidermis development\n",
      "query_prot GO:0001501 0.00049 skeletal system development\n",
      "query_prot GO:0009612 0.00048 response to mechanical stimulus\n",
      "query_prot GO:0007169 0.00048 transmembrane receptor protein tyrosine kinase signaling pathway\n",
      "query_prot GO:0016570 0.00048 histone modification\n",
      "query_prot GO:0032409 0.00048 regulation of transporter activity\n",
      "query_prot GO:0048701 0.00046 embryonic cranial skeleton morphogenesis\n",
      "query_prot GO:0033500 0.00046 carbohydrate homeostasis\n",
      "query_prot GO:0051047 0.00046 positive regulation of secretion\n",
      "query_prot GO:0051249 0.00046 regulation of lymphocyte activation\n",
      "query_prot GO:0007409 0.00046 axonogenesis\n",
      "query_prot GO:0045137 0.00045 development of primary sexual characteristics\n",
      "query_prot GO:0016055 0.00045 Wnt signaling pathway\n",
      "query_prot GO:0001101 0.00044 response to acid chemical\n",
      "query_prot GO:0002694 0.00044 regulation of leukocyte activation\n",
      "query_prot GO:0009415 0.00044 response to water\n",
      "query_prot GO:0060271 0.00044 cilium assembly\n",
      "query_prot GO:0062197 0.00043 cellular response to chemical stress\n",
      "query_prot GO:0060538 0.00043 skeletal muscle organ development\n",
      "query_prot GO:0050920 0.00043 regulation of chemotaxis\n",
      "query_prot GO:0032496 0.00042 response to lipopolysaccharide\n",
      "query_prot GO:0034765 0.00042 regulation of monoatomic ion transmembrane transport\n",
      "query_prot GO:0016042 0.00042 lipid catabolic process\n",
      "query_prot GO:0007611 0.00041 learning or memory\n",
      "query_prot GO:0042440 0.00040 pigment metabolic process\n",
      "query_prot GO:1903706 0.00040 regulation of hemopoiesis\n",
      "query_prot GO:1903522 0.00040 regulation of blood circulation\n",
      "query_prot GO:0046879 0.00040 hormone secretion\n",
      "query_prot GO:0061448 0.00040 connective tissue development\n",
      "query_prot GO:0050769 0.00038 positive regulation of neurogenesis\n",
      "query_prot GO:2001233 0.00037 regulation of apoptotic signaling pathway\n",
      "query_prot GO:0007498 0.00035 mesoderm development\n",
      "query_prot GO:1901607 0.00034 alpha-amino acid biosynthetic process\n",
      "query_prot GO:0097485 0.00033 neuron projection guidance\n",
      "query_prot GO:0007613 0.00031 memory\n",
      "query_prot GO:0009306 0.00030 protein secretion\n",
      "query_prot GO:0044550 0.00029 secondary metabolite biosynthetic process\n",
      "query_prot GO:0002790 0.00029 peptide secretion\n",
      "query_prot GO:0090287 0.00029 regulation of cellular response to growth factor stimulus\n",
      "query_prot GO:0007264 0.00028 small GTPase mediated signal transduction\n",
      "query_prot GO:0048705 0.00026 skeletal system morphogenesis\n",
      "query_prot GO:0035592 0.00026 establishment of protein localization to extracellular region\n",
      "query_prot GO:0046661 0.00025 male sex differentiation\n",
      "query_prot GO:1901342 0.00025 regulation of vasculature development\n",
      "query_prot GO:0010001 0.00024 glial cell differentiation\n",
      "query_prot GO:0045765 0.00024 regulation of angiogenesis\n",
      "query_prot GO:0009953 0.00024 dorsal/ventral pattern formation\n",
      "query_prot GO:0060828 0.00023 regulation of canonical Wnt signaling pathway\n",
      "query_prot GO:0090092 0.00022 regulation of transmembrane receptor protein serine/threonine kinase signaling pathway\n",
      "query_prot GO:0050795 0.00022 regulation of behavior\n",
      "query_prot GO:0002791 0.00022 regulation of peptide secretion\n",
      "query_prot GO:0051963 0.00022 regulation of synapse assembly\n",
      "query_prot GO:0007268 0.00021 chemical synaptic transmission\n",
      "query_prot GO:0051703 0.00021 biological process involved in intraspecies interaction between organisms\n",
      "query_prot GO:0050867 0.00021 positive regulation of cell activation\n",
      "query_prot GO:0045088 0.00021 regulation of innate immune response\n",
      "query_prot GO:0001505 0.00020 regulation of neurotransmitter levels\n",
      "query_prot GO:0009414 0.00020 response to water deprivation\n",
      "query_prot GO:0007605 0.00020 sensory perception of sound\n",
      "query_prot GO:0016051 0.00019 carbohydrate biosynthetic process\n",
      "query_prot GO:0016358 0.00018 dendrite development\n",
      "query_prot GO:0050770 0.00018 regulation of axonogenesis\n",
      "query_prot GO:0051235 0.00018 maintenance of location\n",
      "query_prot GO:1904062 0.00017 regulation of monoatomic cation transmembrane transport\n",
      "query_prot GO:0044242 0.00017 cellular lipid catabolic process\n",
      "query_prot GO:0051962 0.00017 positive regulation of nervous system development\n",
      "query_prot GO:0050954 0.00017 sensory perception of mechanical stimulus\n",
      "query_prot GO:1903829 0.00016 positive regulation of protein localization\n",
      "query_prot GO:0002064 0.00015 epithelial cell development\n",
      "query_prot GO:0050863 0.00014 regulation of T cell activation\n",
      "query_prot GO:0048565 0.00013 digestive tract development\n",
      "query_prot GO:0007517 0.00013 muscle organ development\n",
      "query_prot GO:0090558 0.00013 plant epidermis development\n",
      "query_prot GO:0019748 0.00012 secondary metabolic process\n",
      "query_prot GO:0051056 0.00012 regulation of small GTPase mediated signal transduction\n",
      "query_prot GO:0048667 0.00012 cell morphogenesis involved in neuron differentiation\n",
      "query_prot GO:0046942 0.00009 carboxylic acid transport\n",
      "query_prot GO:0009581 0.00009 detection of external stimulus\n",
      "query_prot GO:0009582 0.00009 detection of abiotic stimulus\n",
      "query_prot GO:0015833 0.00009 peptide transport\n",
      "query_prot GO:0042445 0.00008 hormone metabolic process\n",
      "query_prot GO:0032868 0.00006 response to insulin\n",
      "query_prot GO:0005996 0.00003 monosaccharide metabolic process\n",
      "query_prot GO:1903509 0.00000 liposaccharide metabolic process\n",
      "query_prot GO:0001763 0.00000 morphogenesis of a branching structure\n",
      "query_prot GO:0005976 0.00000 polysaccharide metabolic process\n",
      "query_prot GO:0000271 0.00000 polysaccharide biosynthetic process\n"
     ]
    }
   ],
   "source": [
    "# Q9CQV8\n",
    "!python predict.py --seq MTMDKSELVQKAKLAEQAERYDDMAAAMKAVTEQGHELSNEERNLLSVAYKNVVGARRSSWRVISSIEQKTERNEKKQQMGKEYREKIEAELQDICNDVLELLDKYLILNATQAESKVFYLKMKGDYFRYLSEVASGENKQTTVSNSQQAYQEAFEISKKEMQPTHPIRLGLALNFSVFYYEILNSPEKACSLAKTAFDEAIAELDTLNEESYKDSTLIMQLLRDNLTLWTSENQGDEGDAGEGEN -ont bp --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-16 11:51:52.969245: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-16 11:51:54.678248: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-16 11:51:54.870608: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:54.872499: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:0c.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-16 11:51:54.872554: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-16 11:51:54.879970: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-16 11:51:54.880106: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-16 11:51:54.881602: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-16 11:51:54.882022: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-16 11:51:54.886990: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-16 11:51:54.903833: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-16 11:51:54.904138: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-16 11:51:54.904580: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:54.906337: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:54.907925: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-16 11:51:54.908561: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-06-16 11:51:54.916252: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:54.918084: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:0c.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-16 11:51:54.918234: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:54.919877: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:54.921488: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-16 11:51:54.921555: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-16 11:51:56.040689: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-06-16 11:51:56.040753: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 \n",
      "2023-06-16 11:51:56.040772: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N \n",
      "2023-06-16 11:51:56.041532: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:56.043394: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:56.045003: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 11:51:56.046378: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38286 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:00:0c.0, compute capability: 8.0)\n",
      "### Computing predictions on a single protein...\n",
      "2023-06-16 11:51:57.503316: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-16 11:51:58.370189: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8101\n",
      "2023-06-16 11:51:59.733617: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-16 11:52:00.769098: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-16 11:52:00.937306: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "Protein GO-term/EC-number Score GO-term/EC-number name\n",
      "query_prot GO:0005575 0.99802 cellular_component\n",
      "query_prot GO:0110165 0.99779 cellular anatomical entity\n",
      "query_prot GO:0005622 0.97989 intracellular anatomical structure\n",
      "query_prot GO:0043226 0.91238 organelle\n",
      "query_prot GO:0043227 0.86726 membrane-bounded organelle\n",
      "query_prot GO:0043231 0.81403 intracellular membrane-bounded organelle\n",
      "query_prot GO:0043229 0.80497 intracellular organelle\n",
      "query_prot GO:0032991 0.67914 protein-containing complex\n",
      "query_prot GO:0005737 0.63918 cytoplasm\n",
      "query_prot GO:0043232 0.18738 intracellular non-membrane-bounded organelle\n",
      "query_prot GO:0005634 0.17348 nucleus\n",
      "query_prot GO:0140513 0.13849 nuclear protein-containing complex\n",
      "query_prot GO:0005654 0.10755 nucleoplasm\n",
      "query_prot GO:0031974 0.10294 membrane-enclosed lumen\n",
      "query_prot GO:0043228 0.10090 non-membrane-bounded organelle\n",
      "query_prot GO:0005829 0.09687 cytosol\n",
      "query_prot GO:0016020 0.08423 membrane\n",
      "query_prot GO:0031981 0.07344 nuclear lumen\n",
      "query_prot GO:0070013 0.06270 intracellular organelle lumen\n",
      "query_prot GO:0031975 0.05462 envelope\n",
      "query_prot GO:0043233 0.05327 organelle lumen\n",
      "query_prot GO:1902494 0.01304 catalytic complex\n",
      "### Saving predictions to *.json file...\n"
     ]
    }
   ],
   "source": [
    "# A0A021WW32\n",
    "!python predict.py --seq MFYEHIILAKKGPLARIWLAAHWDKKITKAHVFETNIEKSVEGILQPKVKLALRTSGHLLLGVVRIYSRKAKYLLADCNEAFVKIKMAFRPGMVDLPEGHREANVNAITLPEVFHDFDTALPELNDIDIEAQFSINQSRADEITMREDYGSLSLSLQDDGFGDIGFEAETPEIIRCSIPSNINDKIFDNDVLENIESLDPHSLDAHADMPGSRLDGDGFGDSFGQPALFEDDLFGDPSQPVEQITKESTTVLNADDSDEDAIDNIHNVPSPATSLVNSIEDEKEENNLNGHASVSENVPMNEITLVQNEDEGFALAPLDVSMYKGVTKAKRKRKLIIDEIKNISGEEMKAQLADTSDILTTLDLAPPTKRLMYWKETGGVEKLFSLPSRSIPARALFGNYNRQLFSHSTFFEDFSSVVPMEILALEFYTKENENALIIFNKKGRKRKNDNMSNLFLDHVPDSVVQSLEAPEVLRANHKSLGVSTVSVEIVSKEQESISCQNELTFFDNMRSPDLLSLNEMEQFSSINELPLTPRNMNHEMGDDFNQGDSTPAGLDHGHATPQHGNIGEMDHDSVIPTKKTAVILNESVGTSVLSDNGVSKRTNNILKGWDNYEIPSFVGQGIRHAGHCYQQYH -ont cc --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-06-16 18:03:53.710232: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-16 18:04:00.459152: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1\n",
      "2023-06-16 18:04:00.636507: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:00.639733: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:0c.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-16 18:04:00.639862: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-16 18:04:00.788615: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-16 18:04:00.788786: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-16 18:04:01.013587: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10\n",
      "2023-06-16 18:04:01.031092: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10\n",
      "2023-06-16 18:04:01.102470: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11\n",
      "2023-06-16 18:04:01.155856: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11\n",
      "2023-06-16 18:04:01.158436: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-16 18:04:01.160693: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:01.162460: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:01.169188: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-16 18:04:01.184226: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA\n",
      "To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "2023-06-16 18:04:01.202827: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:01.204584: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: \n",
      "pciBusID: 0000:00:0c.0 name: NVIDIA A100-PCIE-40GB computeCapability: 8.0\n",
      "coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.45GiB deviceMemoryBandwidth: 1.41TiB/s\n",
      "2023-06-16 18:04:01.204748: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:01.206456: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:01.208030: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0\n",
      "2023-06-16 18:04:01.213519: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0\n",
      "2023-06-16 18:04:03.245432: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2023-06-16 18:04:03.245489: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 \n",
      "2023-06-16 18:04:03.245503: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N \n",
      "2023-06-16 18:04:03.250106: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:03.255066: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:03.258407: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:937] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
      "2023-06-16 18:04:03.265321: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 38286 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:00:0c.0, compute capability: 8.0)\n",
      "### Computing predictions from fasta...\n",
      "### Loaded\n",
      "  0%|                                                | 0/141865 [00:00<?, ?it/s]2023-06-16 18:04:07.263923: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8\n",
      "2023-06-16 18:04:09.211734: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8101\n",
      "2023-06-16 18:04:14.458118: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11\n",
      "2023-06-16 18:04:16.389878: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11\n",
      "2023-06-16 18:04:16.660527: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "  2%|▌                                  | 2278/141865 [03:07<1:02:55, 36.97it/s]"
     ]
    }
   ],
   "source": [
    "!python predict.py --fasta_fn testsuperset.fasta -ont mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking in indexes: https://repo.huaweicloud.com/repository/pypi/simple\n",
      "Requirement already satisfied: pandas in /root/miniconda3/lib/python3.8/site-packages (2.0.0)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /root/miniconda3/lib/python3.8/site-packages (from pandas) (2022.7)\n",
      "Collecting numpy>=1.20.3\n",
      "  Using cached https://repo.huaweicloud.com/repository/pypi/packages/9c/ee/77768cade9607687fadbcc1dcbb82dba0554154b3aa641f9c17233ffabe8/numpy-1.24.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /root/miniconda3/lib/python3.8/site-packages (from pandas) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /root/miniconda3/lib/python3.8/site-packages (from pandas) (2021.3)\n",
      "Requirement already satisfied: six>=1.5 in /root/miniconda3/lib/python3.8/site-packages (from python-dateutil>=2.8.2->pandas) (1.15.0)\n",
      "Installing collected packages: numpy\n",
      "  Attempting uninstall: numpy\n",
      "    Found existing installation: numpy 1.18.5\n",
      "    Uninstalling numpy-1.18.5:\n",
      "      Successfully uninstalled numpy-1.18.5\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "tensorflow-gpu 2.5.0 requires numpy~=1.19.2, but you have numpy 1.24.2 which is incompatible.\n",
      "deepfri 0.0.1 requires numpy==1.18.5, but you have numpy 1.24.2 which is incompatible.\u001b[0m\n",
      "Successfully installed numpy-1.24.2\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
      "Looking in indexes: https://repo.huaweicloud.com/repository/pypi/simple\n",
      "Collecting numpy==1.20.3\n",
      "  Using cached https://repo.huaweicloud.com/repository/pypi/packages/a3/9d/5cc11fb882e628d21f8d490f56610f2e6612c9f55ecb25b2b57d4729c5bc/numpy-1.20.3-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.4 MB)\n",
      "Installing collected packages: numpy\n",
      "  Attempting uninstall: numpy\n",
      "    Found existing installation: numpy 1.24.2\n",
      "    Uninstalling numpy-1.24.2:\n",
      "      Successfully uninstalled numpy-1.24.2\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "tensorflow-gpu 2.5.0 requires numpy~=1.19.2, but you have numpy 1.20.3 which is incompatible.\n",
      "deepfri 0.0.1 requires numpy==1.18.5, but you have numpy 1.20.3 which is incompatible.\u001b[0m\n",
      "Successfully installed numpy-1.20.3\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip install pandas\n",
    "!pip install numpy==1.20.3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd \n",
    "# linear = pd.read_csv('submission.tsv',\n",
    "#     sep='\\t')\n",
    "# print('linear read')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "MF = pd.read_csv('DeepFRI_BP_predictions.csv', skiprows=1).iloc[:,[0,1,2]]\n",
    "BP = pd.read_csv('DeepFRI_CC_predictions.csv', skiprows=1).iloc[:,[0,1,2]]\n",
    "CC = pd.read_csv('DeepFRI_MF_predictions.csv', skiprows=1).iloc[:,[0,1,2]]\n",
    "deepfri = pd.concat([MF,BP,CC])\n",
    "del MF,BP,CC\n",
    "deepfri.columns=['Id','GO term','Confidence']\n",
    "print('deepfri read')\n",
    "deepfri.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "deepfri.to_csv('submission.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "deepfri = deepfri.set_index(['Id', 'GO term'])\n",
    "linear = linear.set_index(['Id', 'GO term'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data = pd.merge(linear,deepfri,left_index=True, right_index=True, how='outer', suffixes=('', '_y'))\n",
    "# del linear,deepfri\n",
    "# gc.collect()\n",
    "# print('merged')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Confidence'] = data['Confidence'].fillna(data['Confidence_y'])\n",
    "data = data.reset_index()\n",
    "data = data.iloc[:,[0,1,2]]\n",
    "print('data prepared')\n",
    "data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_csv('submission.tsv', sep='\\t', index=False)\n",
    "print('created submission')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
